# Faecal Microbiota Transplantation (FMT)

## Does FMT change the microbial community over time?

```{r knitr_options_FMT2, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=7, fig.fullwidth=TRUE,
fig.path='Figures/', warning=FALSE,
message=FALSE)
set.seed(2802)
```

```{r data_FMT, comment="", message=FALSE, warning=FALSE, echo=FALSE}
load("data/data_25082025.Rdata")
load("data/objects_25082025.Rdata")
load("data/alpha_25082025.Rdata")
load("data/beta_25082025.Rdata")
```

### Alpha diversity
```{r filter_data5, comment="", message=FALSE, warning=FALSE}
sample_metadata <- sample_metadata %>%
  mutate(treatment=time_point) %>% 
    mutate(treatment=case_when(
    treatment %in% c("Donor1", "Donor2") ~ "Inoculum",
    TRUE ~ treatment
  ))

alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == sample))%>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT1","FMT2") ) 
```

```{r alpha_div_plot_5, comment="", message=FALSE, warning=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = "sample") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral", "phylogenetic"))) %>%
  filter(type=="Cold_intervention" & treatment %in% c("FMT1", "FMT2")) %>% 
  ggplot(aes(y = value, x = treatment, color=treatment, fill=treatment)) +
  geom_boxplot(width = 0.5,alpha=0.5 ,outlier.shape = NA, show.legend = FALSE) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  scale_color_manual(values=c("#76b183", '#40714b50')) +
  scale_fill_manual(values=c("#76b183", '#40714b50')) +
  facet_wrap( ~ metric, scales = "free")+
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size = 12),
    axis.title.y = element_text(size = 14),
    axis.title.x = element_blank(),
    strip.text = element_text(size = 16, lineheight = 0.6)
      )  +
  ylab("Alpha diversity")

```

***Richness***
```{r LMM_rich5, comment="", message=FALSE, warning=FALSE}
Modelq0GLMMNB <- glmer.nb(richness ~ treatment+(1|individual), data = alpha_div_meta)
summary(Modelq0GLMMNB)
```

***Neutral***
```{r LMM_neutral5, comment="", message=FALSE, warning=FALSE}
Model_neutral <- nlme::lme(fixed = neutral ~ treatment, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_neutral)
```

***Phylogenetic***
```{r LMM_phylo5, comment="", message=FALSE, warning=FALSE}
Model_phylo <- nlme::lme(fixed = phylogenetic ~ treatment, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_phylo)
```

### Beta diversity
```{r filter_data_beta5, comment="", message=FALSE, warning=FALSE}
samples_to_keep <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT1", "FMT2")) %>% 
  select(sample) %>% 
  pull()
subset_meta <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT1", "FMT2"))
```

***Richness***
```{r permanova_rich5, comment="", message=FALSE, warning=FALSE}
richness <- as.matrix(beta_q0n$S)
richness <- as.dist(richness[rownames(richness) %in% samples_to_keep,
               colnames(richness) %in% samples_to_keep])

betadisper(richness, subset_meta$treatment) %>% permutest(.) 
```
```{r permanova_rich5_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(richness ~ treatment,
        data = subset_meta %>% arrange(match(sample,labels(richness))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(sample,labels(richness))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

***Neutral***
```{r permanova_neutral5, comment="", message=FALSE, warning=FALSE}
neutral <- as.matrix(beta_q1n$S)
neutral <- as.dist(neutral[rownames(neutral) %in% samples_to_keep,
               colnames(neutral) %in% samples_to_keep])

betadisper(neutral, subset_meta$treatment) %>% permutest(.) 
```
```{r permanova_neutral5_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(neutral ~ treatment,
        data = subset_meta %>% arrange(match(sample,labels(neutral))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(sample,labels(neutral))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

***Phylogenetic***
```{r permanova_phylo5, comment="", message=FALSE, warning=FALSE}
phylo <- as.matrix(beta_q1p$S)
phylo <- as.dist(phylo[rownames(phylo) %in% samples_to_keep,
               colnames(phylo) %in% samples_to_keep])
betadisper(phylo, subset_meta$treatment) %>% permutest(.) 
```
```{r permanova_phylo5_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(phylo ~ treatment,
        data = subset_meta %>% arrange(match(sample,labels(phylo))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(sample,labels(phylo))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

***NMDS***
```{r beta_div_nmds_2_plot, comment="", message=FALSE, warning=FALSE}
richness %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(subset_meta, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = treatment)) +
  geom_point(size = 4) +
  scale_color_manual(name="Time point",values=c("#76b183", '#40714b50')) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    )

neutral %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(subset_meta, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = treatment)) +
  geom_point(size = 4) +
  scale_color_manual(name="Time point",values=c("#76b183", '#40714b50')) +  
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    )

phylo %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(subset_meta, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = treatment)) +
  geom_point(size = 4) +
  scale_color_manual(name="Time point",values=c("#76b183", '#40714b50')) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    )
```


## Do FMT change the microbial community compared to antibiotics and acclimation?
```{r filter_data61, comment="", message=FALSE, warning=FALSE}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == sample))%>%
  filter(type == "Cold_intervention" & treatment %in% c("Antibiotics", "Acclimation","FMT1","FMT2") ) 
```

### Alpha diversity

***Richness*** 
```{r LMM_rich6, comment="", message=FALSE, warning=FALSE}
# Modelq0GLMMNB <- glmer.nb(richness ~ treatment + (1|individual), data = alpha_div_meta)
# summary(Modelq0GLMMNB)
#emmeans(Modelq0GLMMNB, pairwise ~ treatment)

model_nb <- MASS::glm.nb(richness ~ treatment, data = alpha_div_meta)
summary(model_nb)
emmeans(model_nb, pairwise ~ treatment)

```

***Neutral***
```{r filter_neutral_data61, comment="", message=FALSE, warning=FALSE}
Model_neutral <- lme(fixed = neutral ~ treatment, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_neutral)
emmeans(Model_neutral, pairwise ~ treatment)
```

```{r LMM_neutral6, comment="", message=FALSE, warning=FALSE}
Model_neutral <- lme(fixed = neutral ~ treatment, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_neutral)
emmeans(Model_neutral, pairwise ~ treatment)
```

***Phylogenetic***
```{r LMM_phylo6, comment="", message=FALSE, warning=FALSE}
Model_phylo <- lme(fixed = phylogenetic ~ treatment, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_phylo)
emmeans(Model_phylo, pairwise ~ treatment)
```

### Beta diversity
```{r filter_data_beta6, comment="", message=FALSE, warning=FALSE}
samples_to_keep <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("Acclimation", "Antibiotics", "FMT1", "FMT2")) %>% 
  select(sample) %>% 
  pull()
subset_meta <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("Acclimation", "Antibiotics", "FMT1", "FMT2"))
```

***Richness***
```{r permanova_rich6, comment="", message=FALSE, warning=FALSE}
richness <- as.matrix(beta_q0n$S)
richness <- as.dist(richness[rownames(richness) %in% samples_to_keep,
               colnames(richness) %in% samples_to_keep])

betadisper(richness, subset_meta$treatment) %>% permutest(.) 
```

```{r permanova_rich6_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(richness ~ treatment,
        data = subset_meta %>% arrange(match(sample,labels(richness))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(sample,labels(richness))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```
```{r pairwise_rich6, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(richness, subset_meta$treatment, perm = 999)
```

***Neutral***
```{r permanova_neutral6, comment="", message=FALSE, warning=FALSE}
neutral <- as.matrix(beta_q1n$S)
neutral <- as.dist(neutral[rownames(neutral) %in% samples_to_keep,
               colnames(neutral) %in% samples_to_keep])

betadisper(neutral, subset_meta$treatment) %>% permutest(.)
```

```{r permanova_neutral6_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(neutral ~ treatment,
        data = subset_meta %>% arrange(match(sample,labels(neutral))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(sample,labels(neutral))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```
```{r pairwise_neutral6, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(neutral, subset_meta$treatment, perm = 999)
```

***Phylogenetic***
```{r permanova_phylo6, comment="", message=FALSE, warning=FALSE}
phylo <- as.matrix(beta_q1p$S)
phylo <- as.dist(phylo[rownames(phylo) %in% samples_to_keep,
               colnames(phylo) %in% samples_to_keep])
betadisper(phylo, subset_meta$treatment) %>% permutest(.) 
```
```{r permanova_phylo6_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(phylo ~ treatment,
        data = subset_meta %>% arrange(match(sample,labels(phylo))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(sample,labels(phylo))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```
```{r pairwise_phylo6, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(phylo, subset_meta$treatment, perm = 999)
```


## Is the gut microbiota similar to the inoculum after FMT?

### Alpha diversity
```{r alpha_div_plot_7, comment="", message=FALSE, warning=FALSE}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == sample))%>%
  filter(type == "Cold_intervention" & treatment %in% c("Acclimation", "Antibiotics", "Inoculum", "FMT1", "FMT2") )

alpha_div_meta$treatment <- factor(alpha_div_meta$treatment, levels=c("Acclimation","Antibiotics","Inoculum", "FMT1", "FMT2"))

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = "sample") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic"))) %>%
  filter(type=="Cold_intervention" & treatment %in% c("Acclimation", "Antibiotics", "Inoculum", "FMT1", "FMT2")) %>% 
  ggplot(aes(y = value, x = treatment, color=treatment, fill=treatment)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5, outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(name="Time point",values=c('#008080',"#003a45", "#d5b52c", "#76b183",'#40714b50')) +
  scale_fill_manual(name="Time point",values=c('#008080',"#003a45", "#d5b52c", "#76b183",'#40714b50')) +
  facet_wrap(. ~ metric, scales = "free") +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size = 12),
    axis.text.x = element_text(angle = 45, hjust = 1),
    axis.title.y = element_text(size = 14),
    axis.title.x = element_blank(),
    strip.text = element_text(size = 16, lineheight = 0.6)
      )  +
  ylab("Alpha diversity")
```

```{r filter_data_cold|_int, comment="", message=FALSE, warning=FALSE}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == sample))%>%
  filter(type == "Cold_intervention" & treatment %in% c("Inoculum", "FMT1", "FMT2") )
```


***Richness***
```{r LMM_rich7, comment="", message=FALSE, warning=FALSE}
Modelq0GLMMNB <- glmer.nb(richness ~ treatment+(1|individual), data = alpha_div_meta)
summary(Modelq0GLMMNB)
```

***Neutral***
```{r LMM_neutral7, comment="", message=FALSE, warning=FALSE}
Model_neutral <- lme(fixed = neutral ~ treatment, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_neutral)
```

***Phylogenetic***
```{r LMM_phylo7, comment="", message=FALSE, warning=FALSE}
Model_phylo <- lme(fixed = phylogenetic ~ treatment, data = alpha_div_meta,
               random = ~ 1 | individual)
summary(Model_phylo)
```

### Beta diversity
```{r filter_Data7_beta, comment="", message=FALSE, warning=FALSE}
samples_to_keep <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("Inoculum", "FMT1", "FMT2") ) %>% 
  select(sample) %>%
  pull()

subset_meta <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("Inoculum", "FMT1", "FMT2") )
```

***Richness***
```{r permanova_rich7, comment="", message=FALSE, warning=FALSE}
richness <- as.matrix(beta_q0n$S)
richness <- as.dist(richness[rownames(richness) %in% samples_to_keep,
               colnames(richness) %in% samples_to_keep])

betadisper(richness, subset_meta$treatment) %>% permutest(.) 
```
```{r permanova_rich7_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(richness ~ treatment,
        data = subset_meta %>% arrange(match(sample,labels(richness))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(sample,labels(richness))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

```{r pairwise_rich_fmt, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(richness, subset_meta$treatment, perm = 999)
```

***Neutral***
```{r permanova_neutral7, comment="", message=FALSE, warning=FALSE}
neutral <- as.matrix(beta_q1n$S)
neutral <- as.dist(neutral[rownames(neutral) %in% samples_to_keep,
               colnames(neutral) %in% samples_to_keep])

betadisper(neutral, subset_meta$treatment) %>% permutest(.) 
```

```{r permanova_neutral7_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(neutral ~ treatment,
        data = subset_meta %>% arrange(match(sample,labels(neutral))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(sample,labels(neutral))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

```{r pairwise_neutral_fmt, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(neutral, subset_meta$treatment, perm = 999)
```

***Phylogenetic***
```{r permanova_phylo7, comment="", message=FALSE, warning=FALSE}
phylo <- as.matrix(beta_q1p$S)
phylo <- as.dist(phylo[rownames(phylo) %in% samples_to_keep,
               colnames(phylo) %in% samples_to_keep])
betadisper(phylo, subset_meta$treatment) %>% permutest(.) 
```

```{r permanova_phylo7_adonis, comment="", message=FALSE, warning=FALSE}
adonis2(phylo ~ treatment,
        data = subset_meta %>% arrange(match(sample,labels(phylo))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(sample,labels(phylo))) %>% pull(individual)) %>%
        broom::tidy() %>%
        tt()
```

```{r pairwise_phylo_fmt, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(phylo, subset_meta$treatment, perm = 999)
```

***NMDS***
```{r filter_Data_nmds, comment="", message=FALSE, warning=FALSE}
samples_to_keep <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("Acclimation","Antibiotics","Inoculum", "FMT1", "FMT2") ) %>% 
  select(sample) %>%
  pull()

subset_meta <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("Acclimation","Antibiotics","Inoculum", "FMT1", "FMT2") )

alpha_div_meta$treatment <- factor(alpha_div_meta$treatment, levels=c("Acclimation","Antibiotics","Inoculum", "FMT1", "FMT2"))
richness <- as.matrix(beta_q0n$S)
richness <- as.dist(richness[rownames(richness) %in% samples_to_keep,
               colnames(richness) %in% samples_to_keep])
neutral <- as.matrix(beta_q1n$S)
neutral <- as.dist(neutral[rownames(neutral) %in% samples_to_keep,
               colnames(neutral) %in% samples_to_keep])
phylo <- as.matrix(beta_q1p$S)
phylo <- as.dist(phylo[rownames(phylo) %in% samples_to_keep,
               colnames(phylo) %in% samples_to_keep])
```



```{r beta_div_nmds_7_plot, comment="", message=FALSE, warning=FALSE}
richness %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(subset_meta, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = treatment)) +
  geom_point(size = 3) +
  scale_color_manual(name="Time point",values=c('#008080',"#003a45", "#d5b52c", "#76b183",'#40714b50')) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.5, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    )

neutral %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(subset_meta, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = treatment)) +
  geom_point(size = 3) +
  scale_color_manual(name="Time point",values=c('#008080',"#003a45", "#d5b52c", "#76b183",'#40714b50')) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.5, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    )

phylo %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(subset_meta, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = treatment)) +
  geom_point(size = 3) +
  scale_color_manual(name="Time point",values=c('#008080',"#003a45", "#d5b52c", "#76b183",'#40714b50'))+
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.5, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    )
```

#### Differences in the functional capacities
```{r gift_all_three, comment="", message=FALSE, warning=FALSE}
sample_sub <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT1","FMT2", "Inoculum"))

genome_counts_filtered <- genome_counts_filt %>% 
    select(one_of(c("genome",sample_sub$sample)))

GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filtered$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

genome_counts_filtered_row <- genome_counts_filtered %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 

GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_filtered_row,GIFT_db)
```

```{r gitfs_func_fmt_three, comment="", message=FALSE, warning=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  summarise(MCI = mean(value), sd = sd(value))

MCI <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
kruskal.test(value ~ treatment, data=MCI)
```

### Inoculum vs FMT1

#### Structural zeros
```{r struct_zero_fmt_tranp, comment="", message=FALSE, warning=FALSE}
FMT_samples <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment == "FMT1") %>% 
  dplyr::select(sample) %>%
  pull()

Donor_samples <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment =="Inoculum") %>% 
  dplyr::select(sample) %>% pull()

sample_sub <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT1", "Inoculum"))

structural_zeros <- genome_counts_filt %>% 
  select(one_of(c("genome",subset_meta$sample))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_FMT = all(c_across(all_of(FMT_samples)) == 0)) %>% 
   mutate(all_zeros_Donor = all(c_across(all_of(Donor_samples)) == 0)) %>% 
   mutate(average_FMT = mean(c_across(all_of(FMT_samples)), na.rm = TRUE)) %>% 
   mutate(average_Donor = mean(c_across(all_of(Donor_samples)), na.rm = TRUE)) %>% 
   filter(all_zeros_FMT == TRUE || all_zeros_Donor==TRUE)  %>% 
   mutate(present = case_when(
      all_zeros_FMT & !all_zeros_Donor ~ "Inoculum",
      !all_zeros_FMT & all_zeros_Donor ~ "FMT1",
      !all_zeros_FMT & !all_zeros_Donor ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "FMT1", average_FMT, average_Donor)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
```

***Structural zeros in each group***
```{r struc_count_fmt1_tranp, comment="", message=FALSE, warning=FALSE}
fmt <- structural_zeros %>% 
  filter(present=="FMT1") %>% 
  count(phylum, name = "FMT1") %>%
  arrange(desc(FMT1)) 

fmt_f <- structural_zeros %>% 
  filter(present=="FMT1") %>% 
  count(family, name = "FMT1") %>%
  arrange(desc(FMT1)) 
```

```{r total_sum_struc_tranp_fmt1, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="Inoculum") %>% 
  count(phylum, name = "Inoculum") %>%
  arrange(desc(Inoculum)) %>% 
  full_join(.,fmt, by="phylum" ) %>%
  mutate(across(everything(), ~ replace_na(., 0))) %>%
  as.data.frame() %>% 
  summarise(across(where(is.numeric), ~ sum(.x, na.rm = TRUE)))
```

***Phyla to which the structural zeros belong in each group***
```{r struc_count_merged_fmt1_tranp, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="Inoculum") %>% 
  count(phylum, name = "Inoculum") %>%
  arrange(desc(Inoculum)) %>% 
  full_join(.,fmt, by="phylum" ) %>%
  mutate(across(everything(), ~ replace_na(., 0))) %>% 
  tt()
```

***Families to which the structural zeros belong in each group***
```{r total_sum_family_fmt_tranp, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="Inoculum") %>% 
  count(family, name = "Inoculum") %>%
  arrange(desc(Inoculum)) %>% 
  full_join(.,fmt_f, by="family" ) %>%
  mutate(across(everything(), ~ replace_na(., 0))) %>% 
  tt()
```

##### Functional capacities of the structural zeros
```{r gift_struc_fmt_tranp, comment="", message=FALSE, warning=FALSE}
GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% structural_zeros$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

#Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)

#Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

genome_counts_row <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  filter(genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",sample_sub$sample))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>% 
  column_to_rownames(., "genome")

GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_row,GIFT_db)
```

```{r comunity_elem_FMT_Tras, comment="", message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(., sample_metadata[c(9,10)], by=join_by("sample"=="sample"))
```

```{r commun_wilcox_elem_FMT_trans, comment="", message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,treatment), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ treatment)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(9,10)], by = join_by(sample == sample))

difference_table <- element_gift_filt %>%
  select(-sample) %>%
  group_by(treatment) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=FMT1-Inoculum)%>% 
  mutate(group_color = ifelse(Difference <0, "Inoculum","FMT1")) 
```

```{r commun_wilcox_elem_plot_fmt_tranp, comment="", message=FALSE, warning=FALSE, fig.height=12, fig.width=12, fig.fullwidth=TRUE}
difference_table %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(
          breaks=c("FMT1","Inoculum"),
          values=c('#40714b', "#d5b52c")) +
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Microbial Functional Capacity") + 
  ylab("Mean difference")+
  guides(fill=guide_legend(title="Time point"))
```

#### Differential abundance analysis: composition
```{r phyloseq_fmt_tranp, comment="", message=FALSE, warning=FALSE}
phylo_samples <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT1", "Inoculum")) %>% 
  column_to_rownames("sample") %>%
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r ancom_rand_fmt_tranp, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancombc_FMT_Donor_2803 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "treatment",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r save_ancombc_fmt_tranp, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(ancombc_FMT_Donor_2803, file = "data/ancombc_FMT_Donor_2803.Rdata")
```

```{r load_ancombc_fmt_tranp, warning=FALSE, comments="", message=FALSE}
load("data/ancombc_FMT_Donor_2803.Rdata")
```

```{r ancom_rand_fmt1, comment="", message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancombc_FMT_Donor_2803$res %>%
  dplyr::select(taxon, lfc_treatmentFMT1, p_treatmentFMT1) %>%
  filter(p_treatmentFMT1 < 0.05) %>%
  dplyr::arrange(p_treatmentFMT1) %>%
  left_join(., genome_metadata, by = join_by(taxon == genome)) %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_treatmentFMT1)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

***Differential abundance MAGs in each group***
```{r diff_fmt_tran, comment="", message=FALSE, warning=FALSE}
fmt_count <- ancombc_rand_table_mag %>%
  filter(lfc_treatmentFMT1<0)  %>% 
  count(phylum, name = "FMT1") %>%
  arrange(desc(FMT1)) 

ancombc_rand_table_mag %>%
  filter(lfc_treatmentFMT1>0)  %>% 
  count(phylum, name = "Inoculum") %>%
  arrange(desc(Inoculum))  %>% 
  full_join(.,fmt_count, by="phylum")%>%
  mutate(across(where(is.numeric), ~ replace_na(., 0)))
```

```{r total_sum_fmt_tran, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag %>%
  filter(lfc_treatmentFMT1>0)  %>% 
  count(phylum, name = "Inoculum") %>%
  arrange(desc(Inoculum))  %>% 
  full_join(.,fmt_count, by="phylum")%>%
  mutate(across(where(is.numeric), ~ replace_na(., 0))) %>% 
  as.data.frame() %>% 
  summarise(across(where(is.numeric), ~ sum(.x, na.rm = TRUE)))
```

```{r ancombc_rand_plot_phy_fmt_tran, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_treatmentFMT1, y=forcats::fct_reorder(genome,lfc_treatmentFMT1), fill=phylum)) + #forcats::fct_rev()
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```
MAGs enriched in FMT on the left side and in FMT1 on the right side

***Phyla of the significant MAGs***

FMT1
```{r acclim_sign_features_accli_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_treatmentFMT1<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  

ancombc_rand_table_mag%>%
  filter(lfc_treatmentFMT1<0)  %>% 
  select(phylum, family, genus, species)
```

Inoculum
```{r acclim_sign_features_wild_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(lfc_treatmentFMT1>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  

ancombc_rand_table_mag%>%
  filter(lfc_treatmentFMT1>0)  %>% 
  select(phylum, family, genus, species)
```

#### Differences in the functional capacities
```{r gift_all_fmt, comment="", message=FALSE, warning=FALSE}
sample_sub <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT1", "Inoculum"))
genome_counts_filtered <- genome_counts_filt %>% 
    select(one_of(c("genome",sample_sub$sample)))

GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filtered$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

genome_counts_filtered_row <- genome_counts_filtered %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 

GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_filtered_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_filtered_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_filtered_row,GIFT_db)
```

##### Community elements differences:

***MCI***
```{r gitfs_elem_fmt_tran, comment="", message=FALSE, warning=FALSE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  summarise(MCI = mean(value), sd = sd(value))

MCI <- GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
wilcox.test(value ~ treatment, data=MCI)
```
```{r comunity_elem_fmt_tran, comment="", message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(., sample_metadata[c(9,10)], by=join_by("sample"=="sample"))
```

```{r commun_wilcox_elem_fmt_tran, comment="", message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,treatment), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ treatment)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(9,10)], by = join_by(sample == sample))

difference_table <- element_gift_filt %>%
  select(-sample) %>%
  group_by(treatment) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=FMT1-Inoculum)%>% 
  mutate(group_color = ifelse(Difference <0, "Inoculum","FMT1")) 
```

##### Community functions differences

***MCI***
```{r gitfs_func_fmt_tran, comment="", message=FALSE, warning=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  summarise(MCI = mean(value), sd = sd(value))

MCI <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
wilcox.test(value ~ treatment, data=MCI)
```
```{r comunity_func_fmt_tran, comment="", message=FALSE, warning=FALSE}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(9,10)], by="sample")
```

```{r commun_wilcox_func_fmt_tran, comment="", message=FALSE, warning=FALSE}
unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)

significant_functional <- function_gift %>%
    pivot_longer(-c(sample,treatment), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ treatment)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(trait == Code_function))

significant_functional
```

### Inoculum vs FMT2

#### Structural zeros
```{r struct_zero_fmt2_tranp, comment="", message=FALSE, warning=FALSE}
FMT2_samples <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment == "FMT2") %>% 
  dplyr::select(sample) %>%
  pull()

Inoculum_samples <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment =="Inoculum") %>% 
  dplyr::select(sample) %>% pull()

sample_sub <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT2", "Inoculum"))

structural_zeros <- genome_counts_filt %>% 
  select(one_of(c("genome",subset_meta$sample))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>% 
   rowwise() %>% #compute for each row (genome)
   mutate(all_zeros_FMT2 = all(c_across(all_of(FMT2_samples)) == 0)) %>% 
   mutate(all_zeros_Donor = all(c_across(all_of(Donor_samples)) == 0)) %>% 
   mutate(average_FMT2 = mean(c_across(all_of(FMT2_samples)), na.rm = TRUE)) %>% 
   mutate(average_Donor = mean(c_across(all_of(Donor_samples)), na.rm = TRUE)) %>% 
   filter(all_zeros_FMT2 == TRUE || all_zeros_Donor==TRUE)  %>% 
   mutate(present = case_when(
      all_zeros_FMT2 & !all_zeros_Donor ~ "Inoculum",
      !all_zeros_FMT2 & all_zeros_Donor ~ "FMT2",
      !all_zeros_FMT2 & !all_zeros_Donor ~ "None",
      TRUE ~ NA_character_
    )) %>%
   mutate(average = ifelse(present == "FMT2", average_FMT2, average_Donor)) %>%
   dplyr::select(genome, present, average) %>%
   left_join(genome_metadata, by=join_by(genome==genome)) %>%
   arrange(present,-average)
```

***Structural zeros in each group***
```{r struc_count_fmt2_tranp, comment="", message=FALSE, warning=FALSE}
fmt <- structural_zeros %>% 
  filter(present=="FMT2") %>% 
  count(phylum, name = "FMT2") %>%
  arrange(desc(FMT2)) 

fmt2_f <- structural_zeros %>% 
  filter(present=="FMT2") %>% 
  count(family, name = "FMT2") %>%
  arrange(desc(FMT2)) 
```

```{r total_sum_struc_tranp, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="Inoculum") %>% 
  count(phylum, name = "Inoculum") %>%
  arrange(desc(Inoculum)) %>% 
  full_join(.,fmt, by="phylum" ) %>%
  mutate(across(everything(), ~ replace_na(., 0))) %>%
  as.data.frame() %>% 
  summarise(across(where(is.numeric), ~ sum(.x, na.rm = TRUE)))

```

***Phyla to which the structural zeros belong in each group***
```{r struc_count_merged_fmt2_tranp, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="Inoculum") %>% 
  count(phylum, name = "Inoculum") %>%
  arrange(desc(Inoculum)) %>% 
  full_join(.,fmt, by="phylum" ) %>%
  mutate(across(everything(), ~ replace_na(., 0))) %>% 
  tt()
```

***Families to which the structural zeros belong in each group***
```{r total_sum_family_fmt2_tranp, comment="", message=FALSE, warning=FALSE}
structural_zeros %>% 
  filter(present=="Inoculum") %>% 
  count(family, name = "Inoculum") %>%
  arrange(desc(Inoculum)) %>% 
  full_join(.,fmt2_f, by="family" ) %>%
  mutate(across(everything(), ~ replace_na(., 0))) %>% 
  tt()
```

##### Functional capacities of the structural zeros
```{r gift_struc_fmt2_tranp, comment="", message=FALSE, warning=FALSE}
#Aggregate bundle-level GIFTs into the compound level
GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% structural_zeros$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)

#Aggregate element-level GIFTs into the function level
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)

#Aggregate function-level GIFTs into overall Biosynthesis, Degradation and Structural GIFTs
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

genome_counts_row <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  filter(genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",sample_sub$sample))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>% 
  column_to_rownames(., "genome")

GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_row,GIFT_db)
```

```{r comunity_elem_FMT2_Tras, comment="", message=FALSE, warning=FALSE}
element_gift_inoculum <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(., sample_metadata[c(9,10)], by=join_by("sample"=="sample"))
```

```{r commun_wilcox_elem_FMT2_trans, comment="", message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift_inoculum %>%
    pivot_longer(-c(sample,treatment), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ treatment)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_inoculum_t <- element_gift_inoculum  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_inoculum_filt <- subset(element_gift_inoculum_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(9,10)], by = join_by(sample == sample))

difference_table <- element_gift_inoculum_filt %>%
  select(-sample) %>%
  group_by(treatment) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=FMT2-Inoculum)%>% 
  mutate(group_color = ifelse(Difference <0, "Inoculum","FMT2")) 
```

```{r commun_wilcox_elem_plot_fmt2_tranp, comment="", message=FALSE, warning=FALSE, fig.height=12, fig.width=12, fig.fullwidth=TRUE}
difference_table %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(
          breaks=c("FMT2","Inoculum"),
          values=c('#40714b50', "#d5b52c")) +
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        panel.background = element_blank(),
          panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                colour = "grey"))+
  xlab("Microbial Functional Capacity") + 
  ylab("Mean difference")+
  guides(fill=guide_legend(title="Time point"))

```

#### Differential abundance analysis: composition
```{r phyloseq_fmt2_tranp, comment="", message=FALSE, warning=FALSE}
phylo_samples <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT2", "Inoculum")) %>% 
  column_to_rownames("sample") %>%
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r ancom_rand_fmt2_tranp, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancombc_FMT2_Donor_2803 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "treatment",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r save_ancombc_fmt2_tranp, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(ancombc_FMT2_Donor_2803, file = "data/ancombc_FMT2_Donor_2803.Rdata")
```

```{r load_ancombc_fmt2_tranp, warning=FALSE, comments="", message=FALSE}
load("data/ancombc_FMT2_Donor_2803.Rdata")
```

```{r ancom_rand, comment="", message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancombc_FMT2_Donor_2803$res %>%
  dplyr::select(taxon, lfc_treatmentFMT2, p_treatmentFMT2) %>%
  filter(p_treatmentFMT2 < 0.05) %>%
  dplyr::arrange(p_treatmentFMT2) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_treatmentFMT2)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

***Differential abundance MAGs in each group***
```{r diff_fmt2_tran, comment="", message=FALSE, warning=FALSE}
fmt2_count <- ancombc_rand_table_mag %>%
  filter(lfc_treatmentFMT2<0)  %>% 
  count(phylum, name = "FMT2") %>%
  arrange(desc(FMT2)) 

ancombc_rand_table_mag %>%
  filter(lfc_treatmentFMT2>0)  %>% 
  count(phylum, name = "Inoculum") %>%
  arrange(desc(Inoculum))  %>% 
  full_join(.,fmt2_count, by="phylum")%>%
  mutate(across(where(is.numeric), ~ replace_na(., 0)))
```

```{r total_sum_fmt2_tran, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag %>%
  filter(lfc_treatmentFMT2>0)  %>% 
  count(phylum, name = "Inoculum") %>%
  arrange(desc(Inoculum))  %>% 
  full_join(.,fmt2_count, by="phylum")%>%
  mutate(across(where(is.numeric), ~ replace_na(., 0))) %>% 
  as.data.frame() %>% 
  summarise(across(where(is.numeric), ~ sum(.x, na.rm = TRUE)))
```

```{r ancombc_rand_plot_phy_fmt2_tran, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag%>%
      mutate(genome=factor(taxon,levels=ancombc_rand_table_mag$taxon)) %>%
ggplot(., aes(x=lfc_treatmentFMT2, y=forcats::fct_reorder(genome,lfc_treatmentFMT2), fill=phylum)) +
  geom_col() + 
  scale_fill_manual(values=tax_color) + 
  geom_hline(yintercept=0) + 
  theme(panel.background = element_blank(),
        axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 8),
        axis.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 14, face = "bold"),
        legend.position = "right", legend.box = "vertical")+
  xlab("log2FoldChange") + 
  ylab("Species")+
  guides(fill=guide_legend(title="Phylum"))
```
MAGs enriched in donor on the left side and in FMT2 on the right side



#### Differences in the functional capacities
```{r gift_all_fmt2, comment="", message=FALSE, warning=FALSE}
sample_sub <- sample_metadata %>%
  filter(type == "Cold_intervention" & treatment %in% c("FMT2", "Inoculum"))

genome_counts_filtered <- genome_counts_filt %>% 
    select(one_of(c("genome",sample_sub$sample)))

GIFTs_elements <- to.elements(genome_gifts,GIFT_db)
GIFTs_elements_filtered <- GIFTs_elements[rownames(GIFTs_elements) %in% genome_counts_filtered$genome,]
GIFTs_elements_filtered <- as.data.frame(GIFTs_elements_filtered) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
GIFTs_functions <- to.functions(GIFTs_elements_filtered,GIFT_db)
GIFTs_domains <- to.domains(GIFTs_functions,GIFT_db)

genome_counts_filtered_row <- genome_counts_filtered %>%
  mutate_at(vars(-genome),~./sum(.)) %>% 
  column_to_rownames(., "genome") 

GIFTs_elements_community <- to.community(GIFTs_elements_filtered,genome_counts_filtered_row,GIFT_db)
GIFTs_functions_community <- to.community(GIFTs_functions,genome_counts_filtered_row,GIFT_db)
GIFTs_domains_community <- to.community(GIFTs_domains,genome_counts_filtered_row,GIFT_db)
```

##### Community elements differences:

***MCI***
```{r gitfs_elem_fmt2_tran, comment="", message=FALSE, warning=FALSE}
GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  summarise(MCI = mean(value), sd = sd(value))

MCI <- GIFTs_elements_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
wilcox.test(value ~ treatment, data=MCI)
```
```{r comunity_elem_fmt2_tran, comment="", message=FALSE, warning=FALSE}
element_gift <- GIFTs_elements_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  left_join(., sample_metadata[c(1,10)], by=join_by("sample"=="Tube_code"))
```

```{r commun_wilcox_elem_fmt2_tran, comment="", message=FALSE, warning=FALSE}
uniqueGIFT_db<- unique(GIFT_db[c(2,4,5,6)]) %>% unite("Function",Function:Element, sep= "_", remove=FALSE)

significant_elements <- element_gift %>%
    pivot_longer(-c(sample,treatment), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ treatment)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  rownames_to_column(., "Elements")  %>%
  left_join(.,uniqueGIFT_db[c(1,3)],by = join_by(trait == Code_element))

element_gift_t <- element_gift  %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "trait")

element_gift_filt <- subset(element_gift_t, trait %in% significant_elements$trait) %>% 
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric)  %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(9,10)], by = join_by(sample == sample))

difference_table <- element_gift_filt %>%
  select(-sample) %>%
  group_by(treatment) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=FMT2-Inoculum)%>% 
  mutate(group_color = ifelse(Difference <0, "Inoculum","FMT2")) 
```

##### Community functions differences

***MCI***
```{r gitfs_func_fmt2_tran, comment="", message=FALSE, warning=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(treatment) %>%
  summarise(MCI = mean(value), sd = sd(value))

MCI <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) 

shapiro.test(MCI$value)
wilcox.test(value ~ treatment, data=MCI)
```
```{r comunity_func_fmt2_tran, comment="", message=FALSE, warning=FALSE}
function_gift <- GIFTs_functions_community %>% 
  as.data.frame() %>% 
  rownames_to_column(., "sample") %>% 
  merge(., sample_metadata[c(9,10)], by="sample")
```

```{r commun_wilcox_func_fmt2_tran, comment="", message=FALSE, warning=FALSE}
unique_funct_db<- GIFT_db[c(3,4,5)] %>% 
  distinct(Code_function, .keep_all = TRUE)

significant_functional <- function_gift %>%
    pivot_longer(-c(sample,treatment), names_to = "trait", values_to = "value") %>%
    group_by(trait) %>%
    summarise(p_value = wilcox.test(value ~ treatment)$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)%>%
  left_join(.,unique_funct_db[c(1,3)],by = join_by(trait == Code_function))

significant_functional
```



## What is the trend of the microbiota in each type after FMT?

### Alpha diversity

```{r alpha_div_plot_all, comment="", message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
label_map <- c(
  "Control" = "Cold-control", 
  "Treatment" = "Cold-intervention",
  "Hot_control" = "Warm-control",
  "richness" = "Species Richness",
  "neutral" = "Neutral Diversity",
  "phylogenetic" = "Phylogenetic Diversity"
)

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point %in% c("FMT1","Acclimation", "FMT2")) %>%
  mutate(metric = factor(metric, levels = c("richness", "neutral", "phylogenetic"))) %>%
  ggplot(aes(y = value, x = time_point, color = type, fill = type)) +
  geom_boxplot(outlier.shape = NA, show.legend = FALSE) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), alpha = 0.5, show.legend = FALSE) +
  facet_grid(metric ~ type, scales = "free_y", labeller = labeller(metric = label_map, type = label_map))+
#  facet_nested(.~metric+type, labeller = labeller(metric = label_map, type = label_map))+
  scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
      scale_fill_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA50","#d57d2c50","#76b18350")) +
  theme_minimal() +
  theme(axis.text.x=element_text(size=10))+
  labs(x = "Time Point", y = "value")
```

```{r 2}
alpha_div_meta <- alpha_div %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1"|time_point=="Acclimation"|time_point=="FMT2") 
```

***Richness***
```{r alpha_div_data_post7_comparison, comment="", message=FALSE, warning=FALSE}
Modelq0GLMMNB <- glmer.nb(richness ~ type*time_point+(1|individual), data = alpha_div_meta)
summary(Modelq0GLMMNB)
emmeans::emmeans(Modelq0GLMMNB, pairwise ~ type)
emmeans::emmeans(Modelq0GLMMNB, pairwise ~ time_point)
```

***Neutral***
```{r alpha_div_data_1_post7_comparison, comment="", message=FALSE, warning=FALSE}
Modelq1n <- nlme::lme(neutral ~ type*time_point, 
               random = ~ 1 | individual,
               data = alpha_div_meta)
summary(Modelq1n)
emmeans::emmeans(Modelq1n, pairwise ~ type)
emmeans::emmeans(Modelq1n, pairwise ~ time_point)
```

***Phylogenetic***
```{r alpha_div_data_2_post7_comparison, comment="", message=FALSE, warning=FALSE}
Model_phylo <- nlme::lme(phylogenetic ~ type*time_point, 
               random = ~ 1 | individual,
               data = alpha_div_meta)
summary(Model_phylo)
emmeans::emmeans(Model_phylo, pairwise ~ type)
emmeans::emmeans(Model_phylo, pairwise ~ time_point)
```

#### Beta diversity

***Number of samples used***
```{r beta_div_neutral_all_samples_2_comparison, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_post7 <- sample_metadata %>%
  filter(time_point=="FMT1"|time_point=="Acclimation"|time_point=="FMT2") %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_post7 <- sample_metadata %>%
  filter(time_point=="FMT1"|time_point=="Acclimation"|time_point=="FMT2")

subset_meta_post7$time_point<-as.factor(subset_meta_post7$time_point)
subset_meta_post7$type<-as.factor(subset_meta_post7$type)
length(samples_to_keep_post7)
```

***Richness***
```{r beta_richness_permanova_post7_2_comparison, warning=FALSE, comments="", message=FALSE}
richness_post7 <- as.matrix(beta_q0n$S)
richness_post7 <- as.dist(richness_post7[rownames(richness_post7) %in% samples_to_keep_post7,
               colnames(richness_post7) %in% samples_to_keep_post7])
betadisper(richness_post7, subset_meta_post7$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post7_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(richness_post7 ~ type*time_point,
        data = subset_meta_post7 %>% arrange(match(Tube_code,labels(richness_post7))),
        permutations = 999,
        strata = subset_meta_post7 %>% arrange(match(Tube_code,labels(richness_post7))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```

***Neutral***
```{r beta_neutral_permanova_post7_2_comparison, warning=FALSE, comments="", message=FALSE}
neutral_post7 <- as.matrix(beta_q1n$S)
neutral_post7 <- as.dist(neutral_post7[rownames(neutral_post7) %in% samples_to_keep_post7,
               colnames(neutral_post7) %in% samples_to_keep_post7])
betadisper(neutral_post7, subset_meta_post7$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post7_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_post7 ~ type*time_point,
        data = subset_meta_post7 %>% arrange(match(Tube_code,labels(neutral_post7))),
        permutations = 999,
        strata = subset_meta_post7 %>% arrange(match(Tube_code,labels(neutral_post7))) %>% pull(individual),
        by="terms") %>%        
        broom::tidy() %>%
        tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_post7_2_comparison, warning=FALSE, comments="", message=FALSE}
phylo_post7 <- as.matrix(beta_q1p$S)
phylo_post7 <- as.dist(phylo_post7[rownames(phylo_post7) %in% samples_to_keep_post7,
               colnames(phylo_post7) %in% samples_to_keep_post7])
betadisper(phylo_post7, subset_meta_post7$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post7_1_comparison, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_post7 ~ type*time_point,
        data = subset_meta_post7 %>% arrange(match(Tube_code,labels(phylo_post7))),
        permutations = 999,
        strata = subset_meta_post7 %>% arrange(match(Tube_code,labels(phylo_post7))) %>% pull(individual),
          by="terms") %>%        
        broom::tidy() %>%
        tt()
```

***dbRDA***
```{r beta_neutral_nmds_post7_comparison, warning=FALSE, comments="", message=FALSE, results="hide"}
#Richness
cca_ord <- capscale(formula = richness_post7 ~ subset_meta_post7$time_point* subset_meta_post7$type)
CAP_df <- as.data.frame(vegan::scores(cca_ord, display = "sites")) %>%
  rownames_to_column('Tube_code') %>%
  left_join(subset_meta_post7, by = 'Tube_code') %>%
  column_to_rownames('Tube_code')%>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE))

biplot_scores <- as.data.frame(vegan::scores(cca_ord, display = "bp")) %>%
  rownames_to_column("Variable")
biplot_scores$Variable <- recode(biplot_scores$Variable, 
                                 "subset_meta_post7$time_pointFMT1" = "FMT1",
                                 "subset_meta_post7$time_pointFMT2"="FMT2",
                                 "subset_meta_post7$typeHot_control"="Warm-control",
                                 "subset_meta_post7$typeTreatment" = "Cold-intervention",
                                 "subset_meta_post7$time_pointFMT1:subset_meta_post7$typeHot_control"="FMT1 Warm-control",
                                 "subset_meta_post7$time_pointFMT2:subset_meta_post7$typeHot_control"="FMT2 Warm-control",
                                 "subset_meta_post7$time_pointFMT1:subset_meta_post7$typeTreatment" = "FMT1 Cold-intervention",
                                 "subset_meta_post7$time_pointFMT2:subset_meta_post7$typeTreatment" = "FMT2 Cold-intervention")

beta_richness_nmds_post7 <-CAP_df %>%
  group_by(type, time_point) %>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=CAP1,y=CAP2, color=type,shape = time_point)) +
  scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
        scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
  geom_point(size=2) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(aes(x=x_cen, y=y_cen, xend=CAP1, yend=CAP2), alpha=0.2) +
  geom_segment(data = biplot_scores, aes(x = 0, y = 0, xend = CAP1, yend = CAP2),
               inherit.aes = FALSE, arrow = arrow(length = unit(0.2, "cm")), color = "black") +
  geom_text(data = biplot_scores, aes(x = CAP1, y = CAP2, label = Variable),
            inherit.aes = FALSE, color = "black", vjust = -0.5, hjust = 0.5)+
  theme_classic()

#Neutral
cca_ord <- capscale(formula = neutral_post7 ~ subset_meta_post7$time_point* subset_meta_post7$type)
CAP_df <- as.data.frame(vegan::scores(cca_ord, display = "sites")) %>%
  rownames_to_column('Tube_code') %>%
  left_join(subset_meta_post7, by = 'Tube_code') %>%
  column_to_rownames('Tube_code')%>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE))

biplot_scores <- as.data.frame(vegan::scores(cca_ord, display = "bp")) %>%
  rownames_to_column("Variable")
biplot_scores$Variable <- recode(biplot_scores$Variable, 
                                 "subset_meta_post7$time_pointFMT1" = "FMT1",
                                 "subset_meta_post7$time_pointFMT2"="FMT2",
                                 "subset_meta_post7$typeHot_control"="Warm-control",
                                 "subset_meta_post7$typeTreatment" = "Cold-intervention",
                                 "subset_meta_post7$time_pointFMT1:subset_meta_post7$typeHot_control"="FMT1 Warm-control",
                                 "subset_meta_post7$time_pointFMT2:subset_meta_post7$typeHot_control"="FMT2 Warm-control",
                                 "subset_meta_post7$time_pointFMT1:subset_meta_post7$typeTreatment" = "FMT1 Cold-intervention",
                                 "subset_meta_post7$time_pointFMT2:subset_meta_post7$typeTreatment" = "FMT2 Cold-intervention")

beta_neutral_nmds_post7 <- CAP_df %>%
  group_by(type, time_point) %>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=CAP1,y=CAP2, color=type,shape = time_point)) +
  scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
        scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
  geom_point(size=2) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(aes(x=x_cen, y=y_cen, xend=CAP1, yend=CAP2), alpha=0.2) +
  geom_segment(data = biplot_scores, aes(x = 0, y = 0, xend = CAP1, yend = CAP2),
               inherit.aes = FALSE, arrow = arrow(length = unit(0.2, "cm")), color = "black") +
  geom_text(data = biplot_scores, aes(x = CAP1, y = CAP2, label = Variable),
            inherit.aes = FALSE, color = "black", vjust = -0.5, hjust = 0.5)+
  theme_classic()


#Phylogenetic
cca_ord <- capscale(formula = phylo_post7 ~ subset_meta_post7$time_point* subset_meta_post7$type)
CAP_df <- as.data.frame(vegan::scores(cca_ord, display = "sites")) %>%
  rownames_to_column('Tube_code') %>%
  left_join(subset_meta_post7, by = 'Tube_code') %>%
  column_to_rownames('Tube_code')%>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE))

biplot_scores <- as.data.frame(vegan::scores(cca_ord, display = "bp")) %>%
  rownames_to_column("Variable")
biplot_scores$Variable <- recode(biplot_scores$Variable, 
                                 "subset_meta_post7$time_pointFMT1" = "FMT1",
                                 "subset_meta_post7$time_pointFMT2"="FMT2",
                                 "subset_meta_post7$typeHot_control"="Warm-control",
                                 "subset_meta_post7$typeTreatment" = "Cold-intervention",
                                 "subset_meta_post7$time_pointFMT1:subset_meta_post7$typeHot_control"="FMT1 Warm-control",
                                 "subset_meta_post7$time_pointFMT2:subset_meta_post7$typeHot_control"="FMT2 Warm-control",
                                 "subset_meta_post7$time_pointFMT1:subset_meta_post7$typeTreatment" = "FMT1 Cold-intervention",
                                 "subset_meta_post7$time_pointFMT2:subset_meta_post7$typeTreatment" = "FMT2 Cold-intervention")

beta_phylogenetic_nmds_post7 <- CAP_df %>%
  group_by(type, time_point) %>%
  mutate(x_cen = mean(CAP1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(CAP2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(., aes(x=CAP1,y=CAP2, color=type,shape = time_point)) +
  scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
        scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
  geom_point(size=2) +
  geom_hline(yintercept = 0, linetype = "dotted") + 
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_segment(aes(x=x_cen, y=y_cen, xend=CAP1, yend=CAP2), alpha=0.2) +
  geom_segment(data = biplot_scores, aes(x = 0, y = 0, xend = CAP1, yend = CAP2),
               inherit.aes = FALSE, arrow = arrow(length = unit(0.2, "cm")), color = "black") +
  geom_text(data = biplot_scores, aes(x = CAP1, y = CAP2, label = Variable),
            inherit.aes = FALSE, color = "black", vjust = -0.5, hjust = 0.5)+
  theme_classic()
```

```{r beta_neutral_nmds_plot_post7_final, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
ggarrange(beta_richness_nmds_post7, beta_neutral_nmds_post7, beta_phylogenetic_nmds_post7, ncol=3, nrow=1, common.legend = TRUE, legend="right")
```
```{r beta_neutral_nmds_plot_post7_final_1, fig.height=8, fig.width=12, fig.fullwidth=TRUE}
beta_neutral_nmds_post7
```

```{r data_FMT_1, comment="", message=FALSE, warning=FALSE, echo=FALSE}
load("data/data_25082025.Rdata")
load("data/objects_25082025.Rdata")
load("data/alpha_25082025.Rdata")
load("data/beta_25082025.Rdata")
```

## Differences between acclimation and FMT1 across the three experimental groups

### Beta diversity

***CI from acclimation to FMT1***
```{r beta_div_accli_fmt1_ci, comment="", message=FALSE, warning=FALSE}

samples_to_keep <- sample_metadata %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Cold_intervention") %>%
  select(Tube_code) %>% 
  pull()
subset_meta <- sample_metadata %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Cold_intervention")

length(samples_to_keep)
```

```{r beta_div_accli_fmt1_ci_1, comment="", message=FALSE, warning=FALSE}

richness <- as.matrix(beta_q0n$S)
richness <- as.dist(richness[rownames(richness) %in% samples_to_keep,
               colnames(richness) %in% samples_to_keep])
betadisper(richness, subset_meta$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_accli_fmt1_ci_6, comment="", message=FALSE, warning=FALSE}
adonis2(richness ~ time_point,
        data = subset_meta %>% arrange(match(Tube_code,labels(richness))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(Tube_code,labels(richness))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```        

```{r beta_div_accli_fmt1_ci_2, comment="", message=FALSE, warning=FALSE}

neutral <- as.matrix(beta_q1n$S)
neutral <- as.dist(neutral[rownames(neutral) %in% samples_to_keep,
               colnames(neutral) %in% samples_to_keep])
betadisper(neutral, subset_meta$time_point) %>% permutest(., pairwise = TRUE)
```    

```{r beta_div_accli_fmt1_ci_5, comment="", message=FALSE, warning=FALSE}
adonis2(neutral ~ time_point,
        data = subset_meta %>% arrange(match(Tube_code,labels(neutral))),
        strata = subset_meta %>% arrange(match(Tube_code,labels(neutral))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```   

```{r beta_div_accli_fmt1_ci_3, comment="", message=FALSE, warning=FALSE}
phylo <- as.matrix(beta_q1p$S)
phylo <- as.dist(phylo[rownames(phylo) %in% samples_to_keep,
               colnames(phylo) %in% samples_to_keep])
betadisper(phylo, subset_meta$time_point) %>% permutest(., pairwise = TRUE)

```   
```{r beta_div_accli_fmt1_ci_4, comment="", message=FALSE, warning=FALSE}

adonis2(phylo ~ time_point,
        data = subset_meta %>% arrange(match(Tube_code,labels(phylo))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(Tube_code,labels(phylo))) %>% pull(individual),
        by="terms") %>%
        tt()
```


***CC from acclimation to FMT1***
```{r beta_div_accli_fmt1_cc, comment="", message=FALSE, warning=FALSE}

samples_to_keep <- sample_metadata %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Cold_control") %>%
  select(Tube_code) %>% 
  pull()
subset_meta <- sample_metadata %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Cold_control")

length(samples_to_keep)
```

```{r beta_div_accli_fmt1_cc_1, comment="", message=FALSE, warning=FALSE}

richness <- as.matrix(beta_q0n$S)
richness <- as.dist(richness[rownames(richness) %in% samples_to_keep,
               colnames(richness) %in% samples_to_keep])
betadisper(richness, subset_meta$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_accli_fmt1_cc_2, comment="", message=FALSE, warning=FALSE}
adonis2(richness ~ time_point,
        data = subset_meta %>% arrange(match(Tube_code,labels(richness))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(Tube_code,labels(richness))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```        

```{r beta_div_accli_fmt1_cc_3, comment="", message=FALSE, warning=FALSE}

neutral <- as.matrix(beta_q1n$S)
neutral <- as.dist(neutral[rownames(neutral) %in% samples_to_keep,
               colnames(neutral) %in% samples_to_keep])
betadisper(neutral, subset_meta$time_point) %>% permutest(., pairwise = TRUE)
```    

```{r beta_div_accli_fmt1_cc_4, comment="", message=FALSE, warning=FALSE}
adonis2(neutral ~ time_point,
        data = subset_meta %>% arrange(match(Tube_code,labels(neutral))),
        strata = subset_meta %>% arrange(match(Tube_code,labels(neutral))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```   

```{r beta_div_accli_fmt1_cc_5, comment="", message=FALSE, warning=FALSE}
phylo <- as.matrix(beta_q1p$S)
phylo <- as.dist(phylo[rownames(phylo) %in% samples_to_keep,
               colnames(phylo) %in% samples_to_keep])
betadisper(phylo, subset_meta$time_point) %>% permutest(., pairwise = TRUE)

```   
```{r beta_div_accli_fmt1_cc_6, comment="", message=FALSE, warning=FALSE}

adonis2(phylo ~ time_point,
        data = subset_meta %>% arrange(match(Tube_code,labels(phylo))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(Tube_code,labels(phylo))) %>% pull(individual),
        by="terms") %>%
        tt()
```

***WC from acclimation to FMT1***
```{r beta_div_accli_fmt1_wc, comment="", message=FALSE, warning=FALSE}

samples_to_keep <- sample_metadata %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Warm_control") %>%
  select(Tube_code) %>% 
  pull()
subset_meta <- sample_metadata %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Warm_control")

length(samples_to_keep)
```

```{r beta_div_accli_fmt1_wc_1, comment="", message=FALSE, warning=FALSE}

richness <- as.matrix(beta_q0n$S)
richness <- as.dist(richness[rownames(richness) %in% samples_to_keep,
               colnames(richness) %in% samples_to_keep])
betadisper(richness, subset_meta$time_point) %>% permutest(., pairwise = TRUE)
```

```{r beta_div_accli_fmt1_wc_2, comment="", message=FALSE, warning=FALSE}
adonis2(richness ~ time_point,
        data = subset_meta %>% arrange(match(Tube_code,labels(richness))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(Tube_code,labels(richness))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```        

```{r beta_div_accli_fmt1_wc_3, comment="", message=FALSE, warning=FALSE}

neutral <- as.matrix(beta_q1n$S)
neutral <- as.dist(neutral[rownames(neutral) %in% samples_to_keep,
               colnames(neutral) %in% samples_to_keep])
betadisper(neutral, subset_meta$time_point) %>% permutest(., pairwise = TRUE)
```    

```{r beta_div_accli_fmt1_wc_4, comment="", message=FALSE, warning=FALSE}
adonis2(neutral ~ time_point,
        data = subset_meta %>% arrange(match(Tube_code,labels(neutral))),
        strata = subset_meta %>% arrange(match(Tube_code,labels(neutral))) %>% pull(individual),
        by="terms") %>%
        broom::tidy() %>%
        tt()
```   

```{r beta_div_accli_fmt1_wc_5, comment="", message=FALSE, warning=FALSE}
phylo <- as.matrix(beta_q1p$S)
phylo <- as.dist(phylo[rownames(phylo) %in% samples_to_keep,
               colnames(phylo) %in% samples_to_keep])
betadisper(phylo, subset_meta$time_point) %>% permutest(., pairwise = TRUE)

```   
```{r beta_div_accli_fmt1_wc_6, comment="", message=FALSE, warning=FALSE}

adonis2(phylo ~ time_point,
        data = subset_meta %>% arrange(match(Tube_code,labels(phylo))),
        permutations = 999,
        strata = subset_meta %>% arrange(match(Tube_code,labels(phylo))) %>% pull(individual),
        by="terms") %>%
        tt()
```



### Functional differences

***CI from acclimation to FMT1***

```{r comunity_elem_FMT3_cold_1, comment="", message=FALSE, warning=FALSE}
significant_element<-element_gift %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Cold_intervention") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 8)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, time_point),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

```{r comunity_elem_fm2_3, comment="", message=FALSE, warning=FALSE}
element_gift_sig <- element_gift %>%
  select(Tube_code, all_of(intersect(
    significant_element$trait,
    colnames(element_gift)
  ))) %>%
  left_join(., sample_metadata[c(1, 6, 8 )], by = join_by(Tube_code == Tube_code)) %>%
  filter(time_point %in% c("FMT1","Acclimation"))%>%
  filter(type=="Cold_intervention")
  
difference_table <- element_gift_sig %>%
  select(-Tube_code, -type) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-FMT1)%>% 
  mutate(group_color = ifelse(Difference <0,"FMT1", "Acclimation")) 
```

```{r commun_wilcox_elem_plot_fmt2_1, comment="", message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(values=c("#76b183",'#008080')) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

***CC from acclimation to FMT1***

```{r comunity_elem_FMT3_cold, comment="", message=FALSE, warning=FALSE}
significant_element<-element_gift %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Cold_control") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 8)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, time_point),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

```{r comunity_elem_fm2_5, comment="", message=FALSE, warning=FALSE}
element_gift_sig <- element_gift %>%
  select(Tube_code, all_of(intersect(
    significant_element$trait,
    colnames(element_gift)
  ))) %>%
  left_join(., sample_metadata[c(1, 6, 8)], by = join_by(Tube_code == Tube_code)) %>%
  filter(time_point %in% c("FMT1","Acclimation"))%>%
  filter(type=="Cold_control")
  
difference_table <- element_gift_sig %>%
  select(-Tube_code, -type) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Acclimation-FMT1)%>% 
  mutate(group_color = ifelse(Difference <0,"FMT1", "Acclimation")) 
```

```{r commun_wilcox_elem_plot_fmt2_5, comment="", message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(values=c("#4477AA",'#008080')) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

***WC from acclimation to FMT1***

```{r comunity_elem_FMT3_cold_2, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(time_point %in% c("FMT1","Acclimation") & type == "Warm_control") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 8)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, time_point),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

### Differential abundances

#### CI: acclimation vs FMT1
***Structural zeros***

```{r struct_zero_ci_accli_fmt1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
accli_ci_samples <- sample_metadata %>% 
  filter(type == "Cold_intervention") %>% 
  filter(time_point == "Acclimation") %>%
  dplyr::select("Tube_code") %>%
  pull()

fmt1_ci_samples <- sample_metadata %>% 
  filter(type == "Cold_intervention") %>% 
  filter(time_point == "FMT1")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_ci_accli_fmt1 <- sample_metadata%>% 
  filter(time_point %in% c("Acclimation", "FMT1") & type == "Cold_intervention")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_ci_accli_fmt1$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_accli_ci = all(c_across(all_of(accli_ci_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_ci_fmt1 = all(c_across(all_of(fmt1_ci_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_accli_ci = mean(c_across(all_of(accli_ci_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_ci_fmt1 = mean(c_across(all_of(fmt1_ci_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_accli_ci == TRUE | all_zeros_ci_fmt1 == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_accli_ci & !all_zeros_ci_fmt1 ~ "FMT1",!all_zeros_accli_ci & all_zeros_ci_fmt1 ~ "Acclimation",!all_zeros_accli_ci & !all_zeros_ci_fmt1 ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "Acclimation", average_accli_ci, average_ci_fmt1)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r structural_zeros_acci_fmt1_ci}
struc <- structural_zeros %>% 
  filter(present=="Acclimation")%>% 
  count(phylum, name = "Acclimation") %>%
  arrange(desc(Acclimation))
structural_zeros %>% 
  filter(present=="FMT1")%>% 
  count(phylum, name = "FMT1") %>%
  arrange(desc(FMT1)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```

```{r structural_zeros_acci_fmt1_ci_1}

phylo_samples <- sample_metadata %>%
  filter(type == "Cold_intervention" & time_point %in% c("Acclimation", "FMT1") )%>% 
  column_to_rownames("Tube_code") %>% 
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r ancom_rand_pond_accli_fmt1_ci, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_ci_accli_fmt1 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "time_point", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r 4, eval=FALSE }
save(ancom_rand_output_ci_accli_fmt1, file="data/ancom_rand_output_ci_accli_fmt1.RData")
```

```{r 5_1, eval=TRUE}
load("data/ancom_rand_output_ci_accli_fmt1.RData")
```
```{r accli_fmt_ci_data}
ancom_rand_output_ci_accli_fmt1$res %>%
  dplyr::select(taxon, lfc_time_pointFMT1, p_time_pointFMT1) %>%
  filter(p_time_pointFMT1 < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()  
```

```{r ancom_rand_FMT1_accli_ci, comment="", echo=TRUE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_ci_accli_fmt1$res %>%
  dplyr::select(taxon, lfc_time_pointFMT1, p_time_pointFMT1) %>%
  #filter(p_time_pointFMT1 < 0.05) %>%
  dplyr::arrange(p_time_pointFMT1) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_pointFMT1)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT1_accli_ci, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=5, fig.width=6, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_time_pointFMT1 < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_time_pointFMT1, y = -log10(p_time_pointFMT1), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_time_pointFMT1 >= 0.05),
             aes(x = lfc_time_pointFMT1, y = -log10(p_time_pointFMT1)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 4.5)) +  
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_time_pointFMT1), y = 4.5, 
           label = "Acclimation", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_time_pointFMT1), y = 4.5, 
           label = "FMT1", hjust = 1, size = 4, fontface = "bold")

```

***Phyla of the significant MAGs in FMT1***
```{r sign_features_FMT1_accli_ci, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT1 < 0.05) %>%
  filter(lfc_time_pointFMT1>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in acclimation***
```{r sign_features_FMT1_accli_ci_1, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
    filter(p_time_pointFMT1 < 0.05) %>%
  filter(lfc_time_pointFMT1<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### CC: acclimation vs FMT1
***Structural zeros***

```{r struct_zero_cc_accli_fmt1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
accli_cc_samples <- sample_metadata %>% 
  filter(type == "Cold_control") %>% 
  filter(time_point == "Acclimation") %>%
  dplyr::select("Tube_code") %>%
  pull()

fmt1_cc_samples <- sample_metadata %>% 
  filter(type == "Cold_control") %>% 
  filter(time_point == "FMT1")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_cc_accli_fmt1 <- sample_metadata%>% 
  filter(time_point %in% c("Acclimation", "FMT1") & type == "Cold_control")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_cc_accli_fmt1$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_accli_cc = all(c_across(all_of(accli_cc_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_cc_fmt1 = all(c_across(all_of(fmt1_cc_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_accli_cc = mean(c_across(all_of(accli_cc_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_cc_fmt1 = mean(c_across(all_of(fmt1_cc_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_accli_cc == TRUE | all_zeros_cc_fmt1 == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_accli_cc & !all_zeros_cc_fmt1 ~ "FMT1",!all_zeros_accli_cc & all_zeros_cc_fmt1 ~ "Acclimation",!all_zeros_accli_cc & !all_zeros_cc_fmt1 ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "Acclimation", average_accli_cc, average_cc_fmt1)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r structural_zeros_accli_fmt1_cc}
struc <- structural_zeros %>% 
  filter(present=="Acclimation")%>% 
  count(phylum, name = "Acclimation") %>%
  arrange(desc(Acclimation))
structural_zeros %>% 
  filter(present=="FMT1")%>% 
  count(phylum, name = "FMT1") %>%
  arrange(desc(FMT1)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```

```{r structural_zeros_acci_fmt1_cc_1}
phylo_samples <- sample_metadata %>%
  filter(type == "Cold_control" & time_point %in% c("Acclimation", "FMT1") )%>% 
  column_to_rownames("Tube_code") %>% 
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r ancom_rand_pond_accli_fmt1_cc, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_cc_accli_fmt1 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "time_point", 
                  rand_formula = "(1|individual)",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r 4_1 ,cc, eval=FALSE}
save(ancom_rand_output_cc_accli_fmt1, file="data/ancom_rand_output_cc_accli_fmt1.RData")
```

```{r 5_2, eval=TRUE}
load("data/ancom_rand_output_cc_accli_fmt1.RData")
```
```{r, cc_1}
ancom_rand_output_cc_accli_fmt1$res %>%
  dplyr::select(taxon, lfc_time_pointFMT1, p_time_pointFMT1) %>%
  filter(p_time_pointFMT1 < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()  
```

```{r ancom_rand_FMT1_accli_cc, comment="", echo=TRUE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_cc_accli_fmt1$res %>%
  dplyr::select(taxon, lfc_time_pointFMT1, p_time_pointFMT1) %>%
  #filter(p_time_pointFMT1 < 0.05) %>%
  dplyr::arrange(p_time_pointFMT1) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_pointFMT1)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT1_accli_cc, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=5, fig.width=6, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_time_pointFMT1 < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_time_pointFMT1, y = -log10(p_time_pointFMT1), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_time_pointFMT1 >= 0.05),
             aes(x = lfc_time_pointFMT1, y = -log10(p_time_pointFMT1)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 11)) + 
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_time_pointFMT1)+0.2, y = 10, 
           label = "Acclimation", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_time_pointFMT1)-0.2, y = 10, 
           label = "FMT1", hjust = 1, size = 4, fontface = "bold")+
  theme(plot.margin = margin(10, 10, 10, 10), clip = "off")

```

***Phyla of the significant MAGs in FMT1***
```{r sign_features_FMT1_accli_cc, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT1 < 0.05) %>%
  filter(lfc_time_pointFMT1>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in acclimation***
```{r sign_features_FMT1_accli_cc_1, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
    filter(p_time_pointFMT1 < 0.05) %>%
  filter(lfc_time_pointFMT1<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### WC: acclimation vs FMT1
***Structural zeros***

```{r struct_zero_wc_accli_fmt1, comment="", echo=FALSE, message=FALSE, warning=FALSE}
accli_wc_samples <- sample_metadata %>% 
  filter(type == "Warm_control") %>% 
  filter(time_point == "Acclimation") %>%
  dplyr::select("Tube_code") %>%
  pull()

fmt1_wc_samples <- sample_metadata %>% 
  filter(type == "Warm_control") %>% 
  filter(time_point == "FMT1")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_wc_accli_fmt1 <- sample_metadata%>% 
  filter(time_point %in% c("Acclimation", "FMT1") & type == "Warm_control")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_wc_accli_fmt1$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_accli_wc = all(c_across(all_of(accli_wc_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_wc_fmt1 = all(c_across(all_of(fmt1_wc_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_accli_wc = mean(c_across(all_of(accli_wc_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_wc_fmt1 = mean(c_across(all_of(fmt1_wc_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_accli_wc == TRUE | all_zeros_wc_fmt1 == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_accli_wc & !all_zeros_wc_fmt1 ~ "FMT1",!all_zeros_accli_wc & all_zeros_wc_fmt1 ~ "Acclimation",!all_zeros_accli_wc & !all_zeros_wc_fmt1 ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "Acclimation", average_accli_wc, average_wc_fmt1)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r structural_zeros_acci_fmt1_wc}
struc <- structural_zeros %>% 
  filter(present=="Acclimation")%>% 
  count(phylum, name = "Acclimation") %>%
  arrange(desc(Acclimation))
structural_zeros %>% 
  filter(present=="FMT1")%>% 
  count(phylum, name = "FMT1") %>%
  arrange(desc(FMT1)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```


```{r structural_zeros_acci_fmt1_wc_1}

phylo_samples <- sample_metadata %>%
  filter(type == "Warm_control" & time_point %in% c("Acclimation", "FMT1") )%>% 
  column_to_rownames("Tube_code") %>% 
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r ancom_rand_pond_accli_fmt1_wc, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_wc_accli_fmt1 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "time_point",
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r 4_3 ,wc, eval=FALSE}
save(ancom_rand_output_wc_accli_fmt1, file="data/ancom_rand_output_wc_accli_fmt1.RData")
```

```{r 5_3, eval=TRUE}
load("data/ancom_rand_output_wc_accli_fmt1.RData")
```
```{r wc_3}
ancom_rand_output_wc_accli_fmt1$res %>%
  dplyr::select(taxon, lfc_time_pointFMT1, p_time_pointFMT1) %>%
  filter(p_time_pointFMT1 < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()  
```

```{r ancom_rand_FMT1_accli_wc, comment="", echo=TRUE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_wc_accli_fmt1$res %>%
  dplyr::select(taxon, lfc_time_pointFMT1, p_time_pointFMT1) %>%
  #filter(p_time_pointFMT1 < 0.05) %>%
  dplyr::arrange(p_time_pointFMT1) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_pointFMT1)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT1_accli_wc, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=5, fig.width=6, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_time_pointFMT1 < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_time_pointFMT1, y = -log10(p_time_pointFMT1), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_time_pointFMT1 >= 0.05),
             aes(x = lfc_time_pointFMT1, y = -log10(p_time_pointFMT1)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 5)) +  
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_time_pointFMT1), y = 5, 
           label = "FMT1", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_time_pointFMT1), y = 5, 
           label = "Acclimation", hjust = 1, size = 4, fontface = "bold")

```

***Phyla of the significant MAGs in FMT1***
```{r sign_features_FMT1_accli_wc, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT1 < 0.05) %>%
  filter(lfc_time_pointFMT1>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in acclimation***
```{r sign_features_FMT1_accli_wc_1, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT1 < 0.05) %>%
  filter(lfc_time_pointFMT1<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```


## Differences between FMT1 and FMT2 across the three experimental groups

### Functional differences

***CI from FMT1 to FMT2***

```{r comunity_elem_FMT3_cold_6, comment="", message=FALSE, warning=FALSE}
significant_element<-element_gift %>%
  filter(time_point %in% c("FMT2","FMT1") & type == "Cold_intervention") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 8)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, time_point),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

```{r comunity_elem_fm2_2, comment="", message=FALSE, warning=FALSE}
element_gift_sig <- element_gift %>%
  select(Tube_code, all_of(intersect(
    significant_element$trait,
    colnames(element_gift)
  ))) %>%
  left_join(., sample_metadata[c(1, 6, 8)], by = join_by(Tube_code == Tube_code)) %>%
  filter(time_point %in% c("FMT2","FMT1"))%>%
  filter(type=="Cold_intervention")
  
difference_table <- element_gift_sig %>%
  select(-Tube_code, -type) %>%
  group_by(time_point) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=FMT1-FMT2)%>% 
  mutate(group_color = ifelse(Difference <0,"FMT2", "FMT1")) 
```

```{r commun_wilcox_elem_plot_fmt2_2, comment="", message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(values=c('#008080',"#76b183")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

***CC from FMT1 to FMT2***

```{r comunity_elem_FMT3_cold_5, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(time_point %in% c("FMT2","FMT1") & type == "Cold_control") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 8)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, time_point),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

***WC from FMT1 to FMT2***

```{r comunity_elem_FMT3_cold_post6, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(time_point %in% c("FMT2","FMT1") & type == "Warm_control") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 8)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, time_point),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ time_point)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

### Differential abundances

#### CI: FMT1 vs FMT2
***Structural zeros***

```{r struct_zero_ci_fmt1_FMT2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
fmt1_ci_samples <- sample_metadata %>% 
  filter(type == "Cold_intervention") %>% 
  filter(time_point == "FMT1") %>%
  dplyr::select("Tube_code") %>%
  pull()

FMT2_ci_samples <- sample_metadata %>% 
  filter(type == "Cold_intervention") %>% 
  filter(time_point == "FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_ci_fmt1_FMT2 <- sample_metadata%>% 
  filter(time_point %in% c("FMT1", "FMT2") & type == "Cold_intervention")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_ci_fmt1_FMT2$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_fmt1_ci = all(c_across(all_of(fmt1_ci_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_ci_FMT2 = all(c_across(all_of(FMT2_ci_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_fmt1_ci = mean(c_across(all_of(fmt1_ci_samples)), na.rm = TRUE)) %>% # get average genome counts across fmt1
  mutate(average_ci_FMT2 = mean(c_across(all_of(FMT2_ci_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_fmt1_ci == TRUE | all_zeros_ci_FMT2 == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_fmt1_ci & !all_zeros_ci_FMT2 ~ "FMT2",!all_zeros_fmt1_ci & all_zeros_ci_FMT2 ~ "FMT1",!all_zeros_fmt1_ci & !all_zeros_ci_FMT2 ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "FMT1", average_fmt1_ci, average_ci_FMT2)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r structural_zeros_acci_FMT2_ci}
struc <- structural_zeros %>% 
  filter(present=="FMT1")%>% 
  count(phylum, name = "FMT1") %>%
  arrange(desc(FMT1))
structural_zeros %>% 
  filter(present=="FMT2")%>% 
  count(phylum, name = "FMT2") %>%
  arrange(desc(FMT2)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```


```{r structural_zeros_acci_FMT2_ci_1}

phylo_samples <- sample_metadata %>%
  filter(type == "Cold_intervention" & time_point %in% c("FMT1", "FMT2") )%>% 
  column_to_rownames("Tube_code") %>% 
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r ancom_rand_pond_fmt1_FMT2_ci, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_ci_fmt1_FMT2 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "time_point", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r 4_4 , eval=FALSE}
save(ancom_rand_output_ci_fmt1_FMT2, file="data/ancom_rand_output_ci_fmt1_FMT2.RData")
```

```{r 5_4, eval=TRUE}
load("data/ancom_rand_output_ci_fmt1_FMT2.RData")
```
```{r}
ancom_rand_output_ci_fmt1_FMT2$res %>%
  dplyr::select(taxon, lfc_time_pointFMT2, p_time_pointFMT2) %>%
  filter(p_time_pointFMT2 < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()  
```

```{r ancom_rand_FMT2_fmt1_ci, comment="", echo=TRUE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_ci_fmt1_FMT2$res %>%
  dplyr::select(taxon, lfc_time_pointFMT2, p_time_pointFMT2) %>%
  #filter(p_time_pointFMT2 < 0.05) %>%
  dplyr::arrange(p_time_pointFMT2) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_pointFMT2)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT2_fmt1_ci, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=5, fig.width=6, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_time_pointFMT2 < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_time_pointFMT2, y = -log10(p_time_pointFMT2), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_time_pointFMT2 >= 0.05),
             aes(x = lfc_time_pointFMT2, y = -log10(p_time_pointFMT2)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 4)) +  
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_time_pointFMT2), y = 4, 
           label = "FMT1", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_time_pointFMT2), y = 4, 
           label = "FMT2", hjust = 1, size = 4, fontface = "bold")

```

***Phyla of the significant MAGs in FMT2***
```{r sign_features_FMT2_fmt1_ci, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT2 < 0.05) %>%
  filter(lfc_time_pointFMT2>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in FMT1***
```{r sign_features_FMT2_fmt1_ci_1, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT2 < 0.05) %>%
  filter(lfc_time_pointFMT2<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### CC: FMT1 vs FMT2
***Structural zeros***

```{r struct_zero_cc_fmt1_FMT2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
fmt1_cc_samples <- sample_metadata %>% 
  filter(type == "Cold_control") %>% 
  filter(time_point == "FMT1") %>%
  dplyr::select("Tube_code") %>%
  pull()

FMT2_cc_samples <- sample_metadata %>% 
  filter(type == "Cold_control") %>% 
  filter(time_point == "FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_cc_fmt1_FMT2 <- sample_metadata%>% 
  filter(time_point %in% c("FMT1", "FMT2") & type == "Cold_control")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_cc_fmt1_FMT2$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_fmt1_cc = all(c_across(all_of(fmt1_cc_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_cc_FMT2 = all(c_across(all_of(FMT2_cc_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_fmt1_cc = mean(c_across(all_of(fmt1_cc_samples)), na.rm = TRUE)) %>% # get average genome counts across fmt1
  mutate(average_cc_FMT2 = mean(c_across(all_of(FMT2_cc_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_fmt1_cc == TRUE | all_zeros_cc_FMT2 == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_fmt1_cc & !all_zeros_cc_FMT2 ~ "FMT2",!all_zeros_fmt1_cc & all_zeros_cc_FMT2 ~ "FMT1",!all_zeros_fmt1_cc & !all_zeros_cc_FMT2 ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "FMT1", average_fmt1_cc, average_cc_FMT2)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r structural_zeros_acci_FMT2_cc}
struc <- structural_zeros %>% 
  filter(present=="FMT1")%>% 
  count(phylum, name = "FMT1") %>%
  arrange(desc(FMT1))
structural_zeros %>% 
  filter(present=="FMT2")%>% 
  count(phylum, name = "FMT2") %>%
  arrange(desc(FMT2)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```


```{r structural_zeros_acci_FMT2_cc_1}

phylo_samples <- sample_metadata %>%
  filter(type == "Cold_control" & time_point %in% c("FMT1", "FMT2") )%>% 
  column_to_rownames("Tube_code") %>% 
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r ancom_rand_pond_fmt1_FMT2_cc, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_cc_fmt1_FMT2 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "time_point", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r 4_5 ,cc, eval=FALSE}
save(ancom_rand_output_cc_fmt1_FMT2, file="data/ancom_rand_output_cc_fmt1_FMT2.RData")
```

```{r 5_5, eval=TRUE}
load("data/ancom_rand_output_cc_fmt1_FMT2.RData")
```
```{r 5_cc}
ancom_rand_output_cc_fmt1_FMT2$res %>%
  dplyr::select(taxon, lfc_time_pointFMT2, p_time_pointFMT2) %>%
  filter(p_time_pointFMT2 < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()  
```

```{r ancom_rand_FMT2_fmt1_cc, comment="", echo=TRUE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_cc_fmt1_FMT2$res %>%
  dplyr::select(taxon, lfc_time_pointFMT2, p_time_pointFMT2) %>%
  #filter(p_time_pointFMT2 < 0.05) %>%
  dplyr::arrange(p_time_pointFMT2) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_pointFMT2)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT2_fmt1_cc, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=5, fig.width=6, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_time_pointFMT2 < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_time_pointFMT2, y = -log10(p_time_pointFMT2), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_time_pointFMT2 >= 0.05),
             aes(x = lfc_time_pointFMT2, y = -log10(p_time_pointFMT2)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 4)) +  
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_time_pointFMT2), y = 4, 
           label = "FMT1", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_time_pointFMT2), y = 4, 
           label = "FMT2", hjust = 1, size = 4, fontface = "bold")

```

***Phyla of the significant MAGs in FMT2***
```{r sign_features_FMT2_fmt1_cc, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT2 < 0.05) %>%
  filter(lfc_time_pointFMT2>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in FMT1***
```{r sign_features_FMT2_fmt1_cc_1, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT2 < 0.05) %>%
  filter(lfc_time_pointFMT2<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### WC: FMT1 vs FMT2
***Structural zeros***

```{r struct_zero_wc_fmt1_FMT2, comment="", echo=FALSE, message=FALSE, warning=FALSE}
fmt1_wc_samples <- sample_metadata %>% 
  filter(type == "Warm_control") %>% 
  filter(time_point == "FMT1") %>%
  dplyr::select("Tube_code") %>%
  pull()

FMT2_wc_samples <- sample_metadata %>% 
  filter(type == "Warm_control") %>% 
  filter(time_point == "FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_wc_fmt1_FMT2 <- sample_metadata%>% 
  filter(time_point %in% c("FMT1", "FMT2") & type == "Warm_control")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_wc_fmt1_FMT2$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_fmt1_wc = all(c_across(all_of(fmt1_wc_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_wc_FMT2 = all(c_across(all_of(FMT2_wc_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_fmt1_wc = mean(c_across(all_of(fmt1_wc_samples)), na.rm = TRUE)) %>% # get average genome counts across fmt1
  mutate(average_wc_FMT2 = mean(c_across(all_of(FMT2_wc_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_fmt1_wc == TRUE | all_zeros_wc_FMT2 == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_fmt1_wc & !all_zeros_wc_FMT2 ~ "FMT2",!all_zeros_fmt1_wc & all_zeros_wc_FMT2 ~ "FMT1",!all_zeros_fmt1_wc & !all_zeros_wc_FMT2 ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "FMT1", average_fmt1_wc, average_wc_FMT2)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r structural_zeros_acci_FMT2_wc}
struc <- structural_zeros %>% 
  filter(present=="FMT1")%>% 
  count(phylum, name = "FMT1") %>%
  arrange(desc(FMT1))
structural_zeros %>% 
  filter(present=="FMT2")%>% 
  count(phylum, name = "FMT2") %>%
  arrange(desc(FMT2)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```


```{r structural_zeros_acci_FMT2_wc_1}
phylo_samples <- sample_metadata %>%
  filter(type == "Warm_control" & time_point %in% c("FMT1", "FMT2") )%>% 
  column_to_rownames("Tube_code") %>% 
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```

```{r ancom_rand_pond_fmt1_FMT2_wc, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_wc_fmt1_FMT2 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "time_point", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r 4_6 ,wc, eval=FALSE}
save(ancom_rand_output_wc_fmt1_FMT2, file="data/ancom_rand_output_wc_fmt1_FMT2.RData")
```

```{r 5_6, eval=TRUE}
load("data/ancom_rand_output_wc_fmt1_FMT2.RData")
```
```{r 6_wc_1}
ancom_rand_output_wc_fmt1_FMT2$res %>%
  dplyr::select(taxon, lfc_time_pointFMT2, p_time_pointFMT2) %>%
  filter(p_time_pointFMT2 < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()  
```

```{r ancom_rand_FMT2_fmt1_wc, comment="", echo=TRUE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_wc_fmt1_FMT2$res %>%
  dplyr::select(taxon, lfc_time_pointFMT2, p_time_pointFMT2) %>%
  #filter(p_time_pointFMT2 < 0.05) %>%
  dplyr::arrange(p_time_pointFMT2) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_time_pointFMT2)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT2_fmt1_wc, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=5, fig.width=6, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_time_pointFMT2 < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_time_pointFMT2, y = -log10(p_time_pointFMT2), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_time_pointFMT2 >= 0.05),
             aes(x = lfc_time_pointFMT2, y = -log10(p_time_pointFMT2)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 4)) +  
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_time_pointFMT2), y = 4, 
           label = "FMT1", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_time_pointFMT2), y = 4, 
           label = "FMT2", hjust = 1, size = 4, fontface = "bold")

```

***Phyla of the significant MAGs in FMT2***
```{r sign_features_FMT2_fmt1_wc, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT2 < 0.05) %>%
  filter(lfc_time_pointFMT2>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in FMT1***
```{r sign_features_FMT2_fmt1_wc_1, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_time_pointFMT2 < 0.05) %>%
  filter(lfc_time_pointFMT2<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

# What are the differences at FMT1 between groups?

## Is the GM of the intervention group more similar to warm population after one week from FMT?

### Shapiro test

```{r shapiro_fmt1_1, comment="", message=FALSE, warning=FALSE,}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT1") %>% 
  filter(metric=="richness") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT1") %>% 
  filter(metric=="neutral") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT1") %>% 
  filter(metric=="phylogenetic") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
```

### Alpha diversity
```{r alpha_div_plot_5_1, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(method="t.test",size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")
```
***p<0.05 when comparing neutral beta diversity between Cold-control and Warm-control

### Beta diversity

***Number of samples used***
```{r beta_div_neutral_treat1_samples_2, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_post1 <- sample_metadata %>%
  filter(time_point == "FMT1") %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_post1 <- sample_metadata %>%
  filter(time_point == "FMT1")
length(samples_to_keep_post1)
```

***Richness***
```{r beta_richness_permanova_post1_2, warning=FALSE, comments="", message=FALSE}
richness_post1 <- as.matrix(beta_q0n$S)
richness_post1 <- as.dist(richness_post1[rownames(richness_post1) %in% samples_to_keep_post1,
               colnames(richness_post1) %in% samples_to_keep_post1])
betadisper(richness_post1, subset_meta_post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_post1_1, warning=FALSE, comments="", message=FALSE}
adonis2(richness_post1 ~ type,
        data = subset_meta_post1 %>% arrange(match(Tube_code,labels(richness_post1))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_post1_3, warning=FALSE, comments="", message=FALSE}
subset_meta_post1_arrange <- column_to_rownames(subset_meta_post1, "Tube_code")
subset_meta_post1_arrange<-subset_meta_post1_arrange[labels(richness_post1),]

pairwise <- pairwise.adonis(richness_post1, subset_meta_post1_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_post1_2, warning=FALSE, comments="", message=FALSE}
neutral_post1 <- as.matrix(beta_q1n$S)
neutral_post1 <- as.dist(neutral_post1[rownames(neutral_post1) %in% samples_to_keep_post1,
               colnames(neutral_post1) %in% samples_to_keep_post1])
betadisper(neutral_post1, subset_meta_post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_post1_1, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_post1 ~ type,
        data = subset_meta_post1 %>% arrange(match(Tube_code,labels(neutral_post1))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_post1_3, warning=FALSE, comments="", message=FALSE}
subset_meta_post1_arrange <- column_to_rownames(subset_meta_post1, "Tube_code")
subset_meta_post1_arrange<-subset_meta_post1_arrange[labels(neutral_post1),]
pairwise <- pairwise.adonis(neutral_post1, subset_meta_post1_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_post1_2, warning=FALSE, comments="", message=FALSE}
phylo_post1 <- as.matrix(beta_q1p$S)
phylo_post1 <- as.dist(phylo_post1[rownames(phylo_post1) %in% samples_to_keep_post1,
               colnames(phylo_post1) %in% samples_to_keep_post1])
betadisper(phylo_post1, subset_meta_post1$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_post1_1, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_post1 ~ type,
        data = subset_meta_post1 %>% arrange(match(Tube_code,labels(phylo_post1))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_phylopermanova_post1_3, warning=FALSE, comments="", message=FALSE}
subset_meta_post1_arrange <- column_to_rownames(subset_meta_post1, "Tube_code")
subset_meta_post1_arrange<-subset_meta_post1_arrange[labels(phylo_post1),]
pairwise <- pairwise.adonis(phylo_post1, subset_meta_post1_arrange$type, perm=999)
pairwise%>%
  tt()
```

***NMDS***
```{r beta_neutral_nmds_post1, warning=FALSE, comments="", message=FALSE, results="hide"}
richness_post1 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post1, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Cold_control", "Warm_control", "Cold_intervention"),
                       labels=c("Cold_control", "Warm_control", "Cold_intervention"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

neutral_post1 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post1, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Cold_control", "Warm_control", "Cold_intervention"),
                       labels=c("Cold_control", "Warm_control", "Cold_intervention"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

phylo_post1 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_post1, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Cold_control", "Warm_control", "Cold_intervention"),
                       labels=c("Cold_control", "Warm_control", "Cold_intervention"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```

### Differentially abundant MAGs
#### Warm vs Intervention

***Structural zeros***

```{r struct_zero_wild_pop_warm, comment="", echo=FALSE, message=FALSE, warning=FALSE}
warm_samples <- sample_metadata %>% 
  filter(type == "Warm_control") %>% 
  filter(time_point == "FMT1") %>%
  dplyr::select("Tube_code") %>%
  pull()

treatment_samples <- sample_metadata %>% 
  filter(type == "Cold_intervention") %>% 
  filter(time_point == "FMT1")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_wild <- sample_metadata%>% 
  filter(type %in% c("Warm_control", "Cold_intervention") & time_point == "FMT1")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_wild$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_warm = all(c_across(all_of(warm_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_treatment = all(c_across(all_of(treatment_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_warm = mean(c_across(all_of(warm_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_treatment = mean(c_across(all_of(treatment_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_warm == TRUE | all_zeros_treatment == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_warm & !all_zeros_treatment ~ "treatment",!all_zeros_warm & all_zeros_treatment ~ "warm_control",!all_zeros_warm & !all_zeros_treatment ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "warm_control", average_warm, average_treatment)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r}
struc <- structural_zeros %>% 
  filter(present=="warm_control")%>% 
  count(phylum, name = "warm_control") %>%
  arrange(desc(warm_control))
structural_zeros %>% 
  filter(present=="treatment")%>% 
  count(phylum, name = "treatment") %>%
  arrange(desc(treatment)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```


```{r FMT1}
physeq_FMT1 <- subset_samples(physeq_all, type %in% c("Warm_control", "Cold_intervention") & time_point == "FMT1")
physeq_FMT1 <- prune_taxa(taxa_sums(physeq_FMT1)>0, physeq_FMT1)
```
```{r ancom_rand_pond_FMT1, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_fmt1 = ancombc2(data = physeq_FMT1, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "type", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```


```{r, eval=FALSE}
save(ancom_rand_output_fmt1, file="data/ancom_rand_output_fmt1.RData")
```

```{r 1, eval=TRUE}
load("data/ancom_rand_output_fmt1.RData")
```


```{r structural_zeros_fmt1_1}
ancom_rand_output_fmt1$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  filter(p_typeTreatment < 0.05)
```

```{r ancom_rand_FMT1, comment="", echo=TRUE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_FMT1@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_fmt1$res %>%
  dplyr::select(taxon, lfc_typeTreatment, p_typeTreatment) %>%
  #filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeTreatment) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeTreatment)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT1, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=5, fig.width=6, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_typeTreatment < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_typeTreatment, y = -log10(p_typeTreatment), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_typeTreatment >= 0.05),
             aes(x = lfc_typeTreatment, y = -log10(p_typeTreatment)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 4)) +  
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_typeTreatment), y = 4, 
           label = "Cold-intervention", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_typeTreatment), y = 4, 
           label = "Warm-control", hjust = 1, size = 4, fontface = "bold")
```

***Phyla of the significant MAGs in warm control***
```{r sign_features_FMT1_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_typeTreatment < 0.05) %>%
  filter(lfc_typeTreatment>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in intervention group***
```{r sign_features_FMT1_cold, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_typeTreatment < 0.05) %>%
  filter(lfc_typeTreatment<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### Intervention vs cold control

***Structural zeros***

```{r struct_zero_wild_pop_inter, comment="", echo=FALSE, message=FALSE, warning=FALSE}
cold_samples <- sample_metadata %>% 
  filter(type == "Cold_control") %>% 
  filter(time_point == "FMT1") %>%
  dplyr::select("Tube_code") %>%
  pull()

treatment_samples <- sample_metadata %>% 
  filter(type == "Cold_intervention") %>% 
  filter(time_point == "FMT1")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_wild <- sample_metadata%>% 
  filter(type %in% c("Cold_control", "Cold_intervention") & time_point == "FMT1")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_wild$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_cold = all(c_across(all_of(cold_samples)) == 0)) %>% # set true if all samples in cold have zeros
  mutate(all_zeros_treatment = all(c_across(all_of(treatment_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_cold = mean(c_across(all_of(cold_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_treatment = mean(c_across(all_of(treatment_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_cold == TRUE | all_zeros_treatment == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_cold & !all_zeros_treatment ~ "Treatment",!all_zeros_cold & all_zeros_treatment ~ "Cold_control",!all_zeros_cold & !all_zeros_treatment ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "Cold_control", average_cold, average_treatment)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r}
struc <- structural_zeros %>% 
  filter(present=="Cold_control")%>% 
  count(phylum, name = "Cold_control") %>%
  arrange(desc(Cold_control))
structural_zeros %>% 
  filter(present=="Cold_intervention")%>% 
  count(phylum, name = "Cold_intervention") %>%
  arrange(desc(Cold_intervention)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```

```{r}
physeq_FMT1_cold <- subset_samples(physeq_all, type %in% c("Cold_control", "Cold_intervention") & time_point == "FMT1")
physeq_FMT1_cold <- prune_taxa(taxa_sums(physeq_FMT1_cold)>0, physeq_FMT1_cold)
```
```{r ancom_rand_pond_FMT1_cold, comment="", message=FALSE, warning=FALSE, eval=FALSE}
ancom_rand_output_cold = ancombc2(data = physeq_FMT1_cold, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "type", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)
```

```{r }
save(ancom_rand_output_cold, file="data/ancom_rand_output_cold.RData")
```

```{r 3, eval=TRUE}
load("data/ancom_rand_output_cold.RData")
```


```{r 3_1, eval=TRUE}
ancom_rand_output_cold$res %>%
  dplyr::select(taxon, lfc_typeCold_intervention, p_typeCold_intervention) %>%
  filter(p_typeCold_intervention < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,8,9,10) %>% 
  tt()
```

```{r ancom_rand_FMT1_cold, comment="", echo=TRUE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_FMT1_cold@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_cold$res %>%
  dplyr::select(taxon, lfc_typeCold_intervention, p_typeCold_intervention) %>%
  #filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeCold_intervention) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeCold_intervention)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT1_cold, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_typeCold_intervention < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_typeCold_intervention, y = -log10(p_typeCold_intervention), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_typeCold_intervention >= 0.05),
             aes(x = lfc_typeCold_intervention, y = -log10(p_typeCold_intervention)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 2.5)) +  
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_typeCold_intervention), y = 2.5, 
           label = "Cold-control", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_typeCold_intervention), y = 2.5, 
           label = "Cold-intervention", hjust = 1, size = 4, fontface = "bold")
```

***Phyla of the significant MAGs in intervention group***
```{r sign_features_FMT1_int, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_typeCold_intervention < 0.05) %>%
  filter(lfc_typeCold_intervention>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in cold control***
```{r sign_features_FMT1_cold_con, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_typeCold_intervention < 0.05) %>%
  filter(lfc_typeCold_intervention<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### Functional differences
***MCI***
```{r gitfs_elements_tm5, warning=FALSE, comments="", message=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT1") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_functional_tm5_stats_2, warning=FALSE, comments="", message=FALSE}
MCI_tm5_ele <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point == "FMT1")

shapiro.test(MCI_tm5_ele$value)#normality test

res.aov <- aov(value ~ type, data = MCI_tm5_ele)#anova test
summary(res.aov)
```

***Functional differences***

CC and CI
```{r comunity_elem_7_1, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Cold_control", "Cold_intervention") & time_point == "FMT1") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 6)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames()
```

CI and WC
```{r comunity_elem_CI_WC, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Warm_control", "Cold_intervention") &
           time_point == "FMT1") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 6)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

CC and WC
```{r comunity_elem_CC_WC, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Warm_control", "Cold_control") &
           time_point == "FMT1") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 6)], by = "Tube_code") %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

## Is the GM of the intervention group still more similar to warm population after one week from FMT?

### Shapiro test

```{r shapiro_fmt2, comment="", message=FALSE, warning=FALSE,}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT2") %>% 
  filter(metric=="richness") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT2") %>% 
  filter(metric=="neutral") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point =="FMT2") %>% 
  filter(metric=="phylogenetic") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
```

### Alpha diversity
```{r alpha_div_plot_FMT2, comment="", message=FALSE, warning=FALSE, fig.height=4, fig.width=10, fig.fullwidth=TRUE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2") %>%
  mutate(metric=factor(metric,levels=c("richness","neutral","phylogenetic"))) %>%
      ggplot(aes(y = value, x = type, group=type, color=type, fill=type)) +
      geom_boxplot(outlier.shape = NA) +
      geom_jitter(alpha=0.5) +
      scale_color_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA","#d57d2c", "#76b183")) +
      scale_fill_manual(name="Type",
          breaks=c("Control","Hot_control", "Treatment"),
          labels=c("Cold-control","Warm-control", "Cold-intervention"),
          values=c("#4477AA50","#d57d2c50","#76b18350")) +
        scale_x_discrete(labels = c("Control" = "Cold-Cold", "Hot_control" = "Hot-Hot", "Treatment" = "Cold-Hot")) +
      facet_wrap(. ~ metric,scales = "free") +
      coord_cartesian(xlim = c(1, NA)) +
      stat_compare_means(method="t.test",size=3, label.x=.7) +
      theme_classic() +
        theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.title.x = element_blank(),
    axis.title.y = element_text(size=10),
    axis.text.x = element_blank(),
    # Increase plot size
    plot.title = element_text(size = 10),
    axis.text = element_text(size = 8),
    axis.title = element_text(size = 8)
      ) +
  ylab("Alpha diversity")

```
***p<0.05 when comparing richness and neutral beta diversity between Cold-control and Cold-intervention


### Beta diversity

***Number of samples used***
```{r beta_div_neutral_FMT2, results = TRUE, message=FALSE, warning=FALSE, comments=FALSE}
samples_to_keep_FMT2 <- sample_metadata %>%
  filter(time_point == "FMT2") %>% 
  select(Tube_code) %>% 
  pull()
subset_meta_FMT2 <- sample_metadata %>%
  filter(time_point == "FMT2")
length(samples_to_keep_FMT2)
```

***Richness***
```{r beta_richness_permanova_FMT2_2, warning=FALSE, comments="", message=FALSE}
richness_FMT2 <- as.matrix(beta_q0n$S)
richness_FMT2 <- as.dist(richness_FMT2[rownames(richness_FMT2) %in% samples_to_keep_FMT2,
               colnames(richness_FMT2) %in% samples_to_keep_FMT2])
betadisper(richness_FMT2, subset_meta_FMT2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_richness_permanova_FMT2_1, warning=FALSE, comments="", message=FALSE}
adonis2(richness_FMT2 ~ type,
        data = subset_meta_FMT2 %>% arrange(match(Tube_code,labels(richness_FMT2))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_richness_permanova_FMT2_3, warning=FALSE, comments="", message=FALSE}
subset_meta_FMT2_arrange <- column_to_rownames(subset_meta_FMT2, "Tube_code")
subset_meta_FMT2_arrange<-subset_meta_FMT2_arrange[labels(richness_FMT2),]

pairwise <- pairwise.adonis(richness_FMT2, subset_meta_FMT2_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Neutral***
```{r beta_neutral_permanova_FMT2_2, warning=FALSE, comments="", message=FALSE}
neutral_FMT2 <- as.matrix(beta_q1n$S)
neutral_FMT2 <- as.dist(neutral_FMT2[rownames(neutral_FMT2) %in% samples_to_keep_FMT2,
               colnames(neutral_FMT2) %in% samples_to_keep_FMT2])
betadisper(neutral_FMT2, subset_meta_FMT2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_neutral_permanova_FMT2_1, warning=FALSE, comments="", message=FALSE}
adonis2(neutral_FMT2 ~ type,
        data = subset_meta_FMT2 %>% arrange(match(Tube_code,labels(neutral_FMT2))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r beta_neutral_permanova_FMT2_3, warning=FALSE, comments="", message=FALSE}
subset_meta_FMT2_arrange <- column_to_rownames(subset_meta_FMT2, "Tube_code")
subset_meta_FMT2_arrange<-subset_meta_FMT2_arrange[labels(neutral_FMT2),]
pairwise <- pairwise.adonis(neutral_FMT2, subset_meta_FMT2_arrange$type, perm=999)
pairwise%>%
  tt()
```

***Phylogenetic***
```{r beta_phylo_permanova_FMT2_2, warning=FALSE, comments="", message=FALSE}
phylo_FMT2 <- as.matrix(beta_q1p$S)
phylo_FMT2 <- as.dist(phylo_FMT2[rownames(phylo_FMT2) %in% samples_to_keep_FMT2,
               colnames(phylo_FMT2) %in% samples_to_keep_FMT2])
betadisper(phylo_FMT2, subset_meta_FMT2$type) %>% permutest(., pairwise = TRUE)
```

```{r beta_phylo_permanova_FMT2_1, warning=FALSE, comments="", message=FALSE}
adonis2(phylo_FMT2 ~ type,
        data = subset_meta_FMT2 %>% arrange(match(Tube_code,labels(phylo_FMT2))),
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```
```{r beta_phylo_permanova_FMT2_pair, warning=FALSE, comments="", message=FALSE}
subset_meta_FMT2_arrange <- column_to_rownames(subset_meta_FMT2, "Tube_code")
subset_meta_FMT2_arrange<-subset_meta_FMT2_arrange[labels(phylo_FMT2),]
pairwise <- pairwise.adonis(phylo_FMT2, subset_meta_FMT2_arrange$type, perm=999)
pairwise%>%
  tt()
```

***NMDS***
```{r beta_neutral_nmds_FMT2, warning=FALSE, comments="", message=FALSE, results="hide"}
 richness_FMT2 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_FMT2, by = join_by(sample == Tube_code)) %>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Richness beta diversity") +
				theme_classic() +
				theme(legend.position="none")

neutral_FMT2 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_FMT2, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
        scale_color_manual(name="Type",
                       breaks=c("Control", "Hot_control", "Treatment"),
                       labels=c("Cold-Cold", "Hot-Hot", "Cold-Hot"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y = "NMDS2", x="NMDS1 \n Neutral beta diversity") +
				theme_classic() +
				theme(legend.position="none")

phylo_FMT2 %>%
				metaMDS(.,trymax = 500, k=2, verbosity=FALSE, trace = FALSE) %>%
				scores() %>%
				as_tibble(., rownames = "sample") %>%
				left_join(subset_meta_FMT2, by = join_by(sample == Tube_code))%>%
			group_by(type) %>%
			mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
			mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
			ungroup() %>%
			ggplot(., aes(x=NMDS1,y=NMDS2, color=type)) +
				scale_color_manual(name="Type",
                       breaks=c("Cold_control", "Warm_control", "Cold_intervention"),
                       labels=c("Cold_control", "Warm_control", "Cold_intervention"),
                       values=c("#4477AA","#d57d2c","#76b183")) +
				geom_point(size=2) +
				geom_segment(aes(x=x_cen, y=y_cen, xend=NMDS1, yend=NMDS2), alpha=0.2) +
        labs(y= element_blank (), x="NMDS1 \n Phylogenetic beta diversity") +
				theme_classic() +
				theme(legend.position="none")
```


### Differentially abundant MAGs
#### Warm vs Intervention
***Structural zeros***

```{r struct_zero_wild_pop, comment="", echo=FALSE, message=FALSE, warning=FALSE}
warm_samples <- sample_metadata %>% 
  filter(type == "Warm_control") %>% 
  filter(time_point == "FMT2") %>%
  dplyr::select("Tube_code") %>%
  pull()

treatment_samples <- sample_metadata %>% 
  filter(type == "Cold_intervention") %>% 
  filter(time_point == "FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_wild <- sample_metadata%>% 
  filter(type %in% c("Warm_control", "Cold_intervention") & time_point == "FMT2")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_wild$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_warm = all(c_across(all_of(warm_samples)) == 0)) %>% # set true if all samples in warm have zeros
  mutate(all_zeros_treatment = all(c_across(all_of(treatment_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_warm = mean(c_across(all_of(warm_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_treatment = mean(c_across(all_of(treatment_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_warm == TRUE | all_zeros_treatment == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_warm & !all_zeros_treatment ~ "treatment",!all_zeros_warm & all_zeros_treatment ~ "Warm_control",!all_zeros_warm & !all_zeros_treatment ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "Warm_control", average_warm, average_treatment)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r}
struc <- structural_zeros %>% 
  filter(present=="Warm_control")%>% 
  count(phylum, name = "Warm_control") %>%
  arrange(desc(Warm_control))
structural_zeros %>% 
  filter(present=="treatment")%>% 
  count(phylum, name = "treatment") %>%
  arrange(desc(treatment)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```
```{r}
physeq_FMT2 <- subset_samples(physeq_all, type %in% c("Warm_control", "Cold_intervention") & time_point == "FMT2")
physeq_FMT2 <- prune_taxa(taxa_sums(physeq_FMT2)>0, physeq_FMT2)
```
```{r ancom_rand_pond_FMT2, comment="", message=FALSE, warning=FALSE}
ancom_rand_output_fmt2 = ancombc2(data = physeq_FMT2, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "type", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r }
save(ancom_rand_output_fmt2, file="data/ancom_rand_output_fmt2.RData")
```

```{r 5, eval=TRUE}
load("data/ancom_rand_output_fmt2.RData")
```
```{r}
ancom_rand_output_fmt2$res %>%
  dplyr::select(taxon, lfc_typeWarm_control, p_typeWarm_control) %>%
  filter(p_typeWarm_control < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()  
```

```{r ancom_rand_FMT2, comment="", echo=TRUE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_FMT2@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_fmt2$res %>%
  dplyr::select(taxon, lfc_typeWarm_control, p_typeWarm_control) %>%
  #filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeWarm_control) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeWarm_control)

ancombc_rand_table_mag
  
colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT2, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_typeWarm_control < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_typeWarm_control, y = -log10(p_typeWarm_control), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_typeWarm_control >= 0.05),
             aes(x = lfc_typeWarm_control, y = -log10(p_typeWarm_control)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 3.5)) +  
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_typeWarm_control), y = 4, 
           label = "Cold-intervention", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_typeTreatment), y = 4, 
           label = "Warm-control", hjust = 1, size = 4, fontface = "bold")
```

***Phyla of the significant MAGs in intervention control***
```{r sign_features_FMT2_warm, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_typeWarm_control < 0.05) %>%
  filter(lfc_typeWarm_control>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in warm group***
```{r sign_features_FMT2_cold, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_typeWarm_control < 0.05) %>%
  filter(lfc_typeWarm_control<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### Intervention vs cold control
***Structural zeros***

```{r struct_zero_wild_pop_fmt2_int, comment="", echo=FALSE, message=FALSE, warning=FALSE}
cold_samples <- sample_metadata %>% 
  filter(type == "Cold_control") %>% 
  filter(time_point == "FMT2") %>%
  dplyr::select("Tube_code") %>%
  pull()

treatment_samples <- sample_metadata %>% 
  filter(type == "Cold_intervention") %>% 
  filter(time_point == "FMT2")%>%
  dplyr::select("Tube_code") %>%
  pull()

meta_fmt2 <- sample_metadata%>% 
  filter(type %in% c("Cold_control", "Cold_intervention") & time_point == "FMT2")

structural_zeros <- genome_counts_filt %>%
  select(c("genome", meta_fmt2$Tube_code)) %>%
  filter(rowSums(across(where(is.numeric)))!=0) %>% 
  rowwise() %>%
  mutate(all_zeros_cold = all(c_across(all_of(cold_samples)) == 0)) %>% # set true if all samples in cold have zeros
  mutate(all_zeros_treatment = all(c_across(all_of(treatment_samples)) == 0)) %>% # set true if all samples in treatment have zeros
  mutate(average_cold = mean(c_across(all_of(cold_samples)), na.rm = TRUE)) %>% # get average genome counts across accli
  mutate(average_treatment = mean(c_across(all_of(treatment_samples)), na.rm = TRUE)) %>% # get average genome counts across post2
  filter(all_zeros_cold == TRUE | all_zeros_treatment == TRUE)  %>% # filter only genomes with structural zeros
  mutate(present = case_when(all_zeros_cold & !all_zeros_treatment ~ "treatment",!all_zeros_cold & all_zeros_treatment ~ "Cold_control",!all_zeros_cold & !all_zeros_treatment ~ "None",TRUE ~ NA_character_)) %>%
  mutate(average = ifelse(present == "Cold_control", average_cold, average_treatment)) %>%
  dplyr::select(genome, present, average) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  arrange(present, -average)

```

```{r}
struc <- structural_zeros %>% 
  filter(present=="Cold_control")%>% 
  count(phylum, name = "Cold_control") %>%
  arrange(desc(Cold_control))
structural_zeros %>% 
  filter(present=="Cold_intervention")%>% 
  count(phylum, name = "Cold_intervention") %>%
  arrange(desc(Cold_intervention)) %>% 
  full_join(., struc, by="phylum") %>%
  mutate(across(everything(), ~ ifelse(is.na(.), 0, .))) %>% 
  tt()
```
```{r}

phylo_samples <- sample_metadata %>%
  filter(type %in% c("Cold_control", "Cold_intervention") & time_point == "FMT2")%>% 
  column_to_rownames("Tube_code") %>% 
  sample_data()
phylo_genome <- genome_counts_filt %>% 
  filter(!genome %in% structural_zeros$genome) %>% 
  select(one_of(c("genome",rownames(phylo_samples)))) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(genome, where(~ is.numeric(.) && sum(.) > 0)) %>%
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>% 
  filter(genome %in% rownames(phylo_genome)) %>% 
  column_to_rownames("genome") %>% 
  dplyr::select(domain,phylum,class,order,family,genus,species) %>% 
  as.matrix() %>% 
  tax_table()

physeq_genome_filtered <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
```


```{r ancom_rand_pond_FMT2_cold, comment="", message=FALSE, warning=FALSE}
ancom_rand_output_cold_fmt2 = ancombc2(data = physeq_genome_filtered, 
                  assay_name = "counts", 
                  tax_level = NULL,
                  fix_formula = "type", 
                  p_adj_method = "holm", 
                  pseudo_sens = TRUE,
                  prv_cut =0, 
                  lib_cut = 0, 
                  s0_perc = 0.05,
                  group = NULL, 
                  struc_zero = FALSE, 
                  neg_lb = FALSE,
                  alpha = 0.05, 
                  n_cl = 2, 
                  verbose = TRUE,
                  global = FALSE, 
                  pairwise = FALSE, 
                  dunnet = FALSE, 
                  trend = FALSE,
                  iter_control = list(tol = 1e-5, max_iter = 20, verbose = FALSE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100), 
                  trend_control = NULL)

```

```{r 8, eval=TRUE}
save(ancom_rand_output_cold_fmt2, file="data/ancom_rand_output_cold_fmt2.RData")
```

```{r 7, eval=TRUE}
load("data/ancom_rand_output_cold_fmt2.RData")

```
```{r}
ancom_rand_output_cold_fmt2$res %>%
  dplyr::select(taxon, lfc_typeCold_intervention, p_typeCold_intervention) %>%
  filter(p_typeCold_intervention < 0.05) %>% 
  left_join(., genome_metadata, by=join_by("taxon"=="genome")) %>% 
  select(2,3,7,8,9) %>% 
  tt()
```

```{r ancom_rand_FMT2_cold, comment="", echo=FALSE, message=FALSE, warning=FALSE}
taxonomy <- data.frame(physeq_genome_filtered@tax_table) %>%
  rownames_to_column(., "taxon") %>%
  mutate_at(vars(order, phylum, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))

ancombc_rand_table_mag <- ancom_rand_output_cold_fmt2$res %>%
  dplyr::select(taxon, lfc_typeCold_intervention, p_typeCold_intervention) %>%
  #filter(p_typeTreatment < 0.05) %>%
  dplyr::arrange(p_typeCold_intervention) %>%
  merge(., taxonomy, by="taxon") %>%
  mutate_at(vars(phylum, species), ~ str_replace(., "[dpcofgs]__", ""))%>%
  dplyr::arrange(lfc_typeCold_intervention)

colors_alphabetic <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate_at(vars(phylum), ~ str_replace(., "[dpcofgs]__", ""))  %>%
  right_join(taxonomy, by=join_by(phylum == phylum)) %>%
  dplyr::select(phylum, colors) %>%
  mutate(colors = str_c(colors, "80"))  %>% #add 80% alpha
	unique() %>%
	dplyr::arrange(phylum)

tax_table <- as.data.frame(unique(ancombc_rand_table_mag$phylum))
  
colnames(tax_table)[1] <- "phylum"
tax_color <- merge(tax_table, colors_alphabetic, by="phylum")%>%
	dplyr::arrange(phylum) %>%
	dplyr::select(colors) %>%
	pull()
```

```{r ancombc_rand_plot_phy_FMT2_cold, comment="", echo=TRUE, message=FALSE, warning=FALSE, fig.height=10, fig.width=8, fig.fullwidth=TRUE}
ancombc_rand_table_mag %>%
  mutate(
    genome = factor(taxon, levels = ancombc_rand_table_mag$taxon),
    significance = ifelse(p_typeCold_intervention < 0.05, phylum, NA)  # Set non-significant to NA (not mapped in legend)
  ) %>%
  ggplot() +
  # Plot significant points with legend
  geom_point(aes(x = lfc_typeCold_intervention, y = -log10(p_typeCold_intervention), color = significance), 
             size = 3, alpha = 0.7, na.rm = TRUE) +  
  # Plot non-significant points separately without legend
  geom_point(data = filter(ancombc_rand_table_mag, p_typeCold_intervention >= 0.05),
             aes(x = lfc_typeCold_intervention, y = -log10(p_typeCold_intervention)), 
             color = "grey70", size = 3, alpha = 0.7, show.legend = FALSE) +
  scale_color_manual(
    values = tax_color,  # Only keeps phylum colors in legend
    na.translate = FALSE  # Removes NA (non-significant) from legend
  ) + 
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") +  
  geom_vline(xintercept = 0, linetype = "dashed") +
  scale_y_continuous(limits = c(0, 4)) +  
  xlab("log2FoldChange") + 
  ylab("-log10(p-value)") +
  guides(color = guide_legend(title = "Phylum")) +  
  theme_minimal()+ 
  annotate("text", x = min(ancombc_rand_table_mag$lfc_typeTreatment), y = 4, 
           label = "Cold-control", hjust = 0, size = 4, fontface = "bold") +
  annotate("text", x = max(ancombc_rand_table_mag$lfc_typeTreatment), y = 4, 
           label = "Cold-intervention", hjust = 1, size = 4, fontface = "bold")
```

***Phyla of the significant MAGs in intervention group***
```{r sign_features_FMT2_int, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_typeCold_intervention < 0.05) %>%
  filter(lfc_typeCold_intervention>0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

***Phyla of the significant MAGs in cold control***
```{r sign_features_FMT2_cold_con, comment="", message=FALSE, warning=FALSE}
ancombc_rand_table_mag%>%
  filter(p_typeCold_intervention < 0.05) %>%
  filter(lfc_typeCold_intervention<0)  %>% 
  count(phylum, name = "sample_count") %>%
  arrange(desc(sample_count))  
```

#### Functional differences
***MCI***
```{r gitfs_elements_tm5_FMT2_cold, warning=FALSE, comments="", message=FALSE}
GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point=="FMT2") %>%
  group_by(type) %>%
  summarise(MCI = mean(value), sd = sd(value))
```

```{r gitfs_functional_FMT2_cold, warning=FALSE, comments="", message=FALSE}
MCI_FMT2 <- GIFTs_functions_community %>%
  rowMeans() %>%
  as_tibble(., rownames = "sample") %>%
  left_join(sample_metadata, by = join_by(sample == Tube_code)) %>%
  filter(time_point == "FMT2")

shapiro.test(MCI_FMT2$value)

res.aov <- aov(value ~ type, data = MCI_FMT2)
summary(res.aov)
```

***Functional differences***

CC and CI
```{r comunity_elem_FMT2_cold, comment="", message=FALSE, warning=FALSE}
significant_elements<-element_gift %>%
  filter(type %in% c("Cold_control", "Cold_intervention") & time_point == "FMT2") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where( ~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 6)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code, type),
               names_to = "trait",
               values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust = p.adjust(p_value, method = "BH")) %>%
  filter(p_adjust < 0.05) %>%
  remove_rownames() %>% 
  left_join(., uniqueGIFT_db, by=join_by("trait"=="Code_element"))
```

```{r comunity_elem_fm2_1, comment="", message=FALSE, warning=FALSE}
element_gift_sig <- element_gift %>%
  select(Tube_code, all_of(intersect(
    significant_elements$trait,
    colnames(element_gift)
  ))) %>%
  left_join(., sample_metadata[c(1, 6)], by = join_by(Tube_code == Tube_code)) %>%
  filter(type!="Warm_control")

difference_table <- element_gift_sig %>%
  select(-Tube_code) %>%
  group_by(type) %>%
  summarise(across(everything(), mean)) %>%
  t() %>%
  row_to_names(row_number = 1) %>%
  as.data.frame() %>%
  mutate_if(is.character, as.numeric) %>%
  rownames_to_column(., "Elements") %>%
  left_join(.,uniqueGIFT_db[c(1,3,4)],by = join_by(Elements == Code_element)) %>% 
  arrange(Function) %>% 
  mutate(Difference=Cold_control-Cold_intervention)%>% 
  mutate(group_color = ifelse(Difference <0, "Cold_control","Cold_intervention")) 
```

```{r commun_wilcox_elem_plot_fmt2, comment="", message=FALSE, warning=FALSE, fig.height=6, fig.width=12, fig.fullwidth=TRUE}
difference_table %>%
  ggplot(aes(x=forcats::fct_reorder(Function,Difference), y=Difference, fill=group_color)) + 
  geom_col() +
  scale_fill_manual(values=c("#4477AA","#76b183")) + 
  geom_hline(yintercept=0) + 
  coord_flip()+
  theme(axis.text = element_text(size = 10),
        axis.title = element_text(size = 12),
        legend.position = "right", 
        legend.title = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_line(size = 0.15, linetype = 'solid',
                                        colour = "grey"))+
  xlab("Function") + 
  ylab("Mean difference")
```

CI and WC
```{r comunity_elem_FMT2_warm, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Warm_control", "Cold_intervention") &
           time_point == "FMT2") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 6)], by = "Tube_code") %>% 
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```

CC and WC
```{r comunity_elem_FMT2_control, comment="", message=FALSE, warning=FALSE}
element_gift %>%
  filter(type %in% c("Warm_control", "Cold_control") &
           time_point == "FMT2") %>%
  select(-time_point, -type) %>%
  filter(rowSums(across(where(is.numeric), ~ . != 0)) > 0) %>%
  select(Tube_code, where(~ is.numeric(.) && sum(.) > 0)) %>%
  left_join(sample_metadata[c(1, 6)], by = "Tube_code") %>%
  pivot_longer(-c(Tube_code,type), names_to = "trait", values_to = "value") %>%
  group_by(trait) %>%
  summarise(p_value = wilcox.test(value ~ type)$p.value) %>%
  mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
  filter(p_adjust < 0.05)%>%
  remove_rownames()
```
